<!DOCTYPE html>
<html xmlns="http://www.w3.org/1999/xhtml" lang="en" xml:lang="en"><head>

<meta charset="utf-8">
<meta name="generator" content="quarto-1.4.537">

<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=yes">

<meta name="author" content="Andrew Ruiz">
<meta name="dcterms.date" content="2024-02-23">

<title>MADA Project Part 2</title>
<style>
code{white-space: pre-wrap;}
span.smallcaps{font-variant: small-caps;}
div.columns{display: flex; gap: min(4vw, 1.5em);}
div.column{flex: auto; overflow-x: auto;}
div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
ul.task-list{list-style: none;}
ul.task-list li input[type="checkbox"] {
  width: 0.8em;
  margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
  vertical-align: middle;
}
/* CSS for syntax highlighting */
pre > code.sourceCode { white-space: pre; position: relative; }
pre > code.sourceCode > span { line-height: 1.25; }
pre > code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
div.sourceCode { margin: 1em 0; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
pre > code.sourceCode { white-space: pre-wrap; }
pre > code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
  }
pre.numberSource { margin-left: 3em;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
pre > code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
</style>


<script src="MADAproject_pt2_ruiz_files/libs/clipboard/clipboard.min.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/quarto-html/quarto.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/quarto-html/popper.min.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/quarto-html/tippy.umd.min.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/quarto-html/anchor.min.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="MADAproject_pt2_ruiz_files/libs/quarto-html/quarto-syntax-highlighting.css" rel="stylesheet" id="quarto-text-highlighting-styles">
<script src="MADAproject_pt2_ruiz_files/libs/bootstrap/bootstrap.min.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/bootstrap/bootstrap-icons.css" rel="stylesheet">
<link href="MADAproject_pt2_ruiz_files/libs/bootstrap/bootstrap.min.css" rel="stylesheet" id="quarto-bootstrap" data-mode="light">
<script src="MADAproject_pt2_ruiz_files/libs/htmlwidgets-1.6.4/htmlwidgets.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/leaflet-1.3.1/leaflet.css" rel="stylesheet">
<script src="MADAproject_pt2_ruiz_files/libs/leaflet-1.3.1/leaflet.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/leafletfix-1.0.0/leafletfix.css" rel="stylesheet">
<script src="MADAproject_pt2_ruiz_files/libs/proj4-2.6.2/proj4.min.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/Proj4Leaflet-1.0.1/proj4leaflet.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/rstudio_leaflet-1.3.1/rstudio_leaflet.css" rel="stylesheet">
<script src="MADAproject_pt2_ruiz_files/libs/leaflet-binding-2.2.1/leaflet.js"></script>
<script src="MADAproject_pt2_ruiz_files/libs/kePrint-0.0.1/kePrint.js"></script>
<link href="MADAproject_pt2_ruiz_files/libs/lightable-0.0.1/lightable.css" rel="stylesheet">


</head>

<body class="fullcontent">

<div id="quarto-content" class="page-columns page-rows-contents page-layout-article">

<main class="content" id="quarto-document-content">

<header id="title-block-header" class="quarto-title-block default">
<div class="quarto-title">
<h1 class="title">MADA Project Part 2</h1>
<p class="subtitle lead">loading, cleaning, exploring the data</p>
</div>



<div class="quarto-title-meta">

    <div>
    <div class="quarto-title-meta-heading">Author</div>
    <div class="quarto-title-meta-contents">
             <p>Andrew Ruiz </p>
          </div>
  </div>
    
    <div>
    <div class="quarto-title-meta-heading">Published</div>
    <div class="quarto-title-meta-contents">
      <p class="date">February 23, 2024</p>
    </div>
  </div>
  
    
  </div>
  


</header>


<section id="introduction" class="level1">
<h1>Introduction</h1>
<section id="this-file-contains-the-process-for-loading-cleaning-and-exploring-the-datasets." class="level2">
<h2 class="anchored" data-anchor-id="this-file-contains-the-process-for-loading-cleaning-and-exploring-the-datasets.">This file contains the process for loading, cleaning and exploring the datasets.</h2>
</section>
</section>
<section id="loading-libraries" class="level1">
<h1>Loading libraries</h1>
</section>
<section id="this-list-contains-the-specific-versions-used-in-the-script." class="level1">
<h1>this list contains the specific versions used in the script.</h1>
<div class="cell">
<div class="sourceCode cell-code" id="cb1"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readxl) <span class="co"># for loading Excel files (version 1.4.0)</span></span>
<span id="cb1-2"><a href="#cb1-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(httr) <span class="co"># for getting data from the web (version 1.4.2)</span></span>
<span id="cb1-3"><a href="#cb1-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr) <span class="co"># for data processing/cleaning (version 1.0.7)</span></span>
<span id="cb1-4"><a href="#cb1-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr) <span class="co"># for data processing/cleaning (version 1.1.3)</span></span>
<span id="cb1-5"><a href="#cb1-5" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(skimr) <span class="co"># for visualizing data (version 2.1.3)</span></span>
<span id="cb1-6"><a href="#cb1-6" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here) <span class="co"># allows relative pathways for data (version 1.0.1)</span></span>
<span id="cb1-7"><a href="#cb1-7" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co"># reading and writing data to and from CSV, TSV, and fixed-width files (version 2.0.1)</span></span>
<span id="cb1-8"><a href="#cb1-8" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(naniar) <span class="co"># tools to work with missing data (version 0.6.1)</span></span>
<span id="cb1-9"><a href="#cb1-9" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2) <span class="co"># plot creation (version 3.3.5)</span></span>
<span id="cb1-10"><a href="#cb1-10" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(sf) <span class="co"># standardized way to encode spatial data (version 1.0-2)</span></span>
<span id="cb1-11"><a href="#cb1-11" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(topicmodels) <span class="co"># tools for topic modeling on text data (version 0.2-12)</span></span>
<span id="cb1-12"><a href="#cb1-12" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(openxlsx) <span class="co"># Facilitates reading from and writing to .xlsx files (version 4.2.4)</span></span>
<span id="cb1-13"><a href="#cb1-13" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lubridate) <span class="co"># Simplifies the work with dates and times (version 1.7.10)</span></span>
<span id="cb1-14"><a href="#cb1-14" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr) <span class="co"># Allows for dynamic report generation in R (version 1.33)</span></span>
<span id="cb1-15"><a href="#cb1-15" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra) <span class="co"># generates simple HTML or LaTeX tables (version 1.3.4)</span></span>
<span id="cb1-16"><a href="#cb1-16" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(webshot2) <span class="co"># tools to take screenshots of web pages (version 0.0.2)</span></span>
<span id="cb1-17"><a href="#cb1-17" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(magick) <span class="co"># advanced image processing in R (version 2.7.3)</span></span>
<span id="cb1-18"><a href="#cb1-18" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(leaflet) <span class="co"># Enables the creation of interactive maps (version 2.0.4.1)</span></span>
<span id="cb1-19"><a href="#cb1-19" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(htmlwidgets) <span class="co"># Facilitates the creation of R bindings to JavaScript libraries (version 1.5.4)</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="loading-data" class="level1">
<h1>Loading data</h1>
<section id="mosquito-trap-data" class="level2">
<h2 class="anchored" data-anchor-id="mosquito-trap-data">Mosquito trap data</h2>
<section id="this-dataset-contains-trap-counts-by-species-for-all-mosquito-traps-set-in-the-bristol-county-from-2007-2023" class="level3">
<h3 class="anchored" data-anchor-id="this-dataset-contains-trap-counts-by-species-for-all-mosquito-traps-set-in-the-bristol-county-from-2007-2023">This dataset contains trap counts by species for all mosquito traps set in the Bristol County from 2007-2023</h3>
</section>
<section id="this-dataset-will-undergo-extensive-processing-below." class="level3">
<h3 class="anchored" data-anchor-id="this-dataset-will-undergo-extensive-processing-below.">this dataset will undergo extensive processing below.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb2"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct the path to the Excel file</span></span>
<span id="cb2-2"><a href="#cb2-2" aria-hidden="true" tabindex="-1"></a><span class="co">#file_path_mos &lt;- here("data", "raw-data", "MOSQ_MADA.xlsx")</span></span>
<span id="cb2-3"><a href="#cb2-3" aria-hidden="true" tabindex="-1"></a>file_path_mos <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"MOSQ_MADA.xlsx"</span>)</span>
<span id="cb2-4"><a href="#cb2-4" aria-hidden="true" tabindex="-1"></a>file_exists_check <span class="ot">&lt;-</span> <span class="fu">file.exists</span>(file_path_mos)</span>
<span id="cb2-5"><a href="#cb2-5" aria-hidden="true" tabindex="-1"></a>file_path_mos <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"raw-data"</span>, <span class="st">"MOSQ_MADA.xlsx"</span>)</span>
<span id="cb2-6"><a href="#cb2-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb2-7"><a href="#cb2-7" aria-hidden="true" tabindex="-1"></a><span class="co">#file_exists_check &lt;- file.exists("/Users/andrewruiz/MADA_course/RUIZ-MADA-project/data/raw-data/MOSQ_MADA.xlsx")</span></span>
<span id="cb2-8"><a href="#cb2-8" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(file_exists_check)  <span class="co"># Should return TRUE if the path is correct</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb4"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb4-1"><a href="#cb4-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(file_path_mos)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "/Users/andrewruiz/MADA_course/RUIZ-MADA-project/data/raw-data/MOSQ_MADA.xlsx"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb6"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(file_exists_check)  <span class="co"># This should return TRUE if the file path is correct</span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] TRUE</code></pre>
</div>
<div class="sourceCode cell-code" id="cb8"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb8-1"><a href="#cb8-1" aria-hidden="true" tabindex="-1"></a><span class="co"># import the data to R and store it in a dataframe named "mosquito"</span></span>
<span id="cb8-2"><a href="#cb8-2" aria-hidden="true" tabindex="-1"></a>mosquito_raw <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path_mos)</span>
<span id="cb8-3"><a href="#cb8-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb8-4"><a href="#cb8-4" aria-hidden="true" tabindex="-1"></a><span class="co">#load the first few rows of data to verify the data loaded as expected</span></span>
<span id="cb8-5"><a href="#cb8-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mosquito_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
  MCD     `Pool ID` `Species code` `Trap Set Date`     `Collection Date`   Town 
  &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;          &lt;dttm&gt;              &lt;dttm&gt;              &lt;chr&gt;
1 Bristol BR21NS-0… PRO            2021-07-26 00:00:00 2021-07-27 00:00:00 Berk…
2 State … SL21NS-6… TVT            2021-08-29 00:00:00 2021-08-30 00:00:00 Rayn…
3 Bristol BR22NS-0… WAL            2022-07-07 00:00:00 2022-07-08 00:00:00 Mans…
4 Bristol BR07-000… ABS            2007-06-04 00:00:00 2007-06-05 00:00:00 Free…
5 Bristol BR07-000… ABS            2007-06-04 00:00:00 2007-06-05 00:00:00 Free…
6 Bristol BR07-000… ABS            2007-06-06 00:00:00 2007-06-07 00:00:00 Free…
# ℹ 6 more variables: `Number of Traps` &lt;dbl&gt;, `Trap Type` &lt;chr&gt;,
#   `Pool Size` &lt;dbl&gt;, `Submitted for Testing` &lt;chr&gt;, Genus &lt;chr&gt;,
#   Species &lt;chr&gt;</code></pre>
</div>
</div>
</section>
</section>
<section id="mosquito-trap-sites" class="level2">
<h2 class="anchored" data-anchor-id="mosquito-trap-sites">Mosquito trap sites</h2>
<section id="location-for-the-mosquito-trap-sites" class="level3">
<h3 class="anchored" data-anchor-id="location-for-the-mosquito-trap-sites">Location for the mosquito trap sites</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb10"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb10-1"><a href="#cb10-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct the path to the file</span></span>
<span id="cb10-2"><a href="#cb10-2" aria-hidden="true" tabindex="-1"></a>file_path_trap <span class="ot">=</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"BCMCP_trap_sites_2021.xlsx"</span>)</span>
<span id="cb10-3"><a href="#cb10-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-4"><a href="#cb10-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import the data to R and store it in a dataframe named "trap_location"</span></span>
<span id="cb10-5"><a href="#cb10-5" aria-hidden="true" tabindex="-1"></a>trap_location <span class="ot">=</span> <span class="fu">read_excel</span>(file_path_trap)</span>
<span id="cb10-6"><a href="#cb10-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb10-7"><a href="#cb10-7" aria-hidden="true" tabindex="-1"></a><span class="co">#load the first few rows of data to verify the data loaded as expected</span></span>
<span id="cb10-8"><a href="#cb10-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(trap_location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 8
  `Trap Site Address`         MCD   Location DecLat DecLong Quadrant Inact Edit 
  &lt;chr&gt;                       &lt;chr&gt; &lt;chr&gt;     &lt;dbl&gt;   &lt;dbl&gt; &lt;chr&gt;    &lt;lgl&gt; &lt;lgl&gt;
1 Crown Street, BR            Bris… Attlebo…   41.9   -71.4 &lt;NA&gt;     NA    NA   
2 Rutledge st., BR            Bris… Attlebo…   41.9   -71.4 &lt;NA&gt;     NA    NA   
3 Adamsdale Road &amp; Siesta Dr… Bris… North A…   41.9   -71.4 &lt;NA&gt;     NA    NA   
4 Lee's Playground            Bris… Attlebo…   41.9   -71.4 &lt;NA&gt;     NA    NA   
5 Newport Avenue #960, BR     Bris… Attlebo…   41.9   -71.4 &lt;NA&gt;     NA    NA   
6 66 Hillside Road, BR        Bris… North A…   42.0   -71.3 &lt;NA&gt;     NA    NA   </code></pre>
</div>
<div class="sourceCode cell-code" id="cb12"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb12-1"><a href="#cb12-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(trap_location)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [152 × 8] (S3: tbl_df/tbl/data.frame)
 $ Trap Site Address: chr [1:152] "Crown Street, BR" "Rutledge st., BR" "Adamsdale Road &amp; Siesta Drive, BR" "Lee's Playground" ...
 $ MCD              : chr [1:152] "Bristol" "Bristol" "Bristol" "Bristol" ...
 $ Location         : chr [1:152] "Attleboro: Attleboro 02703(Bristol)" "Attleboro: Attleboro 02703(Bristol)" "North Attleborough: North Attleborough unknown(Bristol)" "Attleboro: Attleboro 02703(Bristol)" ...
 $ DecLat           : num [1:152] 41.9 41.9 41.9 41.9 41.9 ...
 $ DecLong          : num [1:152] -71.4 -71.4 -71.4 -71.4 -71.4 ...
 $ Quadrant         : chr [1:152] NA NA NA NA ...
 $ Inact            : logi [1:152] NA NA NA NA NA NA ...
 $ Edit             : logi [1:152] NA NA NA NA NA NA ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb14"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb14-1"><a href="#cb14-1" aria-hidden="true" tabindex="-1"></a><span class="co">#The last 3 columns have missing data. However, they are not important for the purposes of this project</span></span>
<span id="cb14-2"><a href="#cb14-2" aria-hidden="true" tabindex="-1"></a><span class="co"># there is not a need to delete the records</span></span>
<span id="cb14-3"><a href="#cb14-3" aria-hidden="true" tabindex="-1"></a><span class="co">#However, we will delete the columns</span></span>
<span id="cb14-4"><a href="#cb14-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb14-5"><a href="#cb14-5" aria-hidden="true" tabindex="-1"></a>trap_location_selected <span class="ot">&lt;-</span> trap_location <span class="sc">%&gt;%</span></span>
<span id="cb14-6"><a href="#cb14-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(<span class="st">`</span><span class="at">Trap Site Address</span><span class="st">`</span>, MCD, Location, DecLat, DecLong)</span>
<span id="cb14-7"><a href="#cb14-7" aria-hidden="true" tabindex="-1"></a>trap_location_selected</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 152 × 5
   `Trap Site Address`               MCD     Location             DecLat DecLong
   &lt;chr&gt;                             &lt;chr&gt;   &lt;chr&gt;                 &lt;dbl&gt;   &lt;dbl&gt;
 1 Crown Street, BR                  Bristol Attleboro: Attlebor…   41.9   -71.4
 2 Rutledge st., BR                  Bristol Attleboro: Attlebor…   41.9   -71.4
 3 Adamsdale Road &amp; Siesta Drive, BR Bristol North Attleborough:…   41.9   -71.4
 4 Lee's Playground                  Bristol Attleboro: Attlebor…   41.9   -71.4
 5 Newport Avenue #960, BR           Bristol Attleboro: Attlebor…   41.9   -71.4
 6 66 Hillside Road, BR              Bristol North Attleborough:…   42.0   -71.3
 7 Ray Hodde st., BR                 Bristol Seekonk: Seekonk 02…   41.8   -71.3
 8 County Street #290, BR            Bristol Seekonk: Seekonk 02…   41.8   -71.3
 9 JL Dietsch Blvd, BR               Bristol North Attleborough:…   42.0   -71.3
10 Barneyville Road, BR              Bristol Swansea: Swansea 02…   41.8   -71.3
# ℹ 142 more rows</code></pre>
</div>
<div class="sourceCode cell-code" id="cb16"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb16-1"><a href="#cb16-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the trap locations selected dataset RDS</span></span>
<span id="cb16-2"><a href="#cb16-2" aria-hidden="true" tabindex="-1"></a>trap_location_selected_path_rds <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"trap_sites_selected.rds"</span>)</span>
<span id="cb16-3"><a href="#cb16-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb16-4"><a href="#cb16-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the trap_location_selected dataframe as an RDS file</span></span>
<span id="cb16-5"><a href="#cb16-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(trap_location_selected, trap_location_selected_path_rds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="mosquito-virus-data" class="level2">
<h2 class="anchored" data-anchor-id="mosquito-virus-data">Mosquito virus data</h2>
<section id="the-virus-isolation-data-is-stored-in-separate-spreadsheets-by-year-from-2014-to-2020" class="level3">
<h3 class="anchored" data-anchor-id="the-virus-isolation-data-is-stored-in-separate-spreadsheets-by-year-from-2014-to-2020">The virus isolation data is stored in separate spreadsheets by year from 2014 to 2020</h3>
</section>
<section id="this-code-will-load-all-the-files-combine-them-and-then-save-the-combined-file-in-the-raw-data-folder" class="level3">
<h3 class="anchored" data-anchor-id="this-code-will-load-all-the-files-combine-them-and-then-save-the-combined-file-in-the-raw-data-folder">This code will load all the files, combine them, and then save the combined file in the “raw data” folder</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb17"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb17-1"><a href="#cb17-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Define the path to the folder with Excel files</span></span>
<span id="cb17-2"><a href="#cb17-2" aria-hidden="true" tabindex="-1"></a>folder_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"virus_data"</span>)</span>
<span id="cb17-3"><a href="#cb17-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-4"><a href="#cb17-4" aria-hidden="true" tabindex="-1"></a><span class="co">#List all Excel files (.xls and .xlsx)</span></span>
<span id="cb17-5"><a href="#cb17-5" aria-hidden="true" tabindex="-1"></a>file_paths <span class="ot">&lt;-</span> <span class="fu">list.files</span>(<span class="at">path =</span> folder_path, <span class="at">pattern =</span> <span class="st">"</span><span class="sc">\\</span><span class="st">.xls[x]*$"</span>, <span class="at">full.names =</span> <span class="cn">TRUE</span>)</span>
<span id="cb17-6"><a href="#cb17-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-7"><a href="#cb17-7" aria-hidden="true" tabindex="-1"></a><span class="co">#Read each file into a list of data frames</span></span>
<span id="cb17-8"><a href="#cb17-8" aria-hidden="true" tabindex="-1"></a>data_list <span class="ot">&lt;-</span> <span class="fu">lapply</span>(file_paths, read_excel)</span>
<span id="cb17-9"><a href="#cb17-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb17-10"><a href="#cb17-10" aria-hidden="true" tabindex="-1"></a><span class="co">#Check the names of the files read</span></span>
<span id="cb17-11"><a href="#cb17-11" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">basename</span>(file_paths))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "2014virus.xls"  "2015virus.xls"  "2016virus.xls"  "2017virus.xls" 
[5] "2018virus.xlsx" "2019virus.xlsx" "2020virus.xlsx"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb19"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb19-1"><a href="#cb19-1" aria-hidden="true" tabindex="-1"></a><span class="co">#Combine the datasets</span></span>
<span id="cb19-2"><a href="#cb19-2" aria-hidden="true" tabindex="-1"></a>combined_virus_data <span class="ot">&lt;-</span> <span class="fu">bind_rows</span>(data_list)</span>
<span id="cb19-3"><a href="#cb19-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-4"><a href="#cb19-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the combined dataset</span></span>
<span id="cb19-5"><a href="#cb19-5" aria-hidden="true" tabindex="-1"></a>combined_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"combined_virus_data.csv"</span>)</span>
<span id="cb19-6"><a href="#cb19-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-7"><a href="#cb19-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the combined data frame as a CSV file</span></span>
<span id="cb19-8"><a href="#cb19-8" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(combined_virus_data, combined_path, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb19-9"><a href="#cb19-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-10"><a href="#cb19-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the combined dataset RDS</span></span>
<span id="cb19-11"><a href="#cb19-11" aria-hidden="true" tabindex="-1"></a>combined_path_rds <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"combined_virus_data.rds"</span>)</span>
<span id="cb19-12"><a href="#cb19-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-13"><a href="#cb19-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the combined data frame as an RDS file</span></span>
<span id="cb19-14"><a href="#cb19-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(combined_virus_data, combined_path_rds)</span>
<span id="cb19-15"><a href="#cb19-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-16"><a href="#cb19-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb19-17"><a href="#cb19-17" aria-hidden="true" tabindex="-1"></a><span class="co"># View the first few rows to confirm it's loaded correctly</span></span>
<span id="cb19-18"><a href="#cb19-18" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(combined_virus_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 5
  County  Town        `Collection Date`   Species           Virus
  &lt;chr&gt;   &lt;chr&gt;       &lt;dttm&gt;              &lt;chr&gt;             &lt;chr&gt;
1 Bristol Freetown    2014-07-31 00:00:00 Culiseta melanura EEE  
2 Bristol Freetown    2014-07-31 00:00:00 Culiseta melanura EEE  
3 Bristol New Bedford 2014-08-04 00:00:00 Culiseta melanura EEE  
4 Bristol Freetown    2014-08-08 00:00:00 Culiseta melanura EEE  
5 Bristol Westport    2014-08-11 00:00:00 Culiseta melanura EEE  
6 Bristol Freetown    2014-08-15 00:00:00 Culiseta melanura EEE  </code></pre>
</div>
</div>
</section>
</section>
<section id="bird-data" class="level2">
<h2 class="anchored" data-anchor-id="bird-data">Bird data</h2>
<section id="the-complete-bird-dataset-is-over-600-mb.-because-it-is-too-large-for-github-it-is-stored-locally.-it-contain-variables-and-records-that-are-not-needed-for-this-analysis.-eight-of-50-varibles-will-be-kept-and-the-records-will-be-filtered-to-show-only-birds-from-the-genus-corvus-observed-in-massachusetts-after-2014-01-01" class="level3">
<h3 class="anchored" data-anchor-id="the-complete-bird-dataset-is-over-600-mb.-because-it-is-too-large-for-github-it-is-stored-locally.-it-contain-variables-and-records-that-are-not-needed-for-this-analysis.-eight-of-50-varibles-will-be-kept-and-the-records-will-be-filtered-to-show-only-birds-from-the-genus-corvus-observed-in-massachusetts-after-2014-01-01">The complete bird dataset is over 600 MB. Because it is too large for GitHub, it is stored locally. It contain variables and records that are not needed for this analysis. Eight of 50 varibles will be kept and the records will be filtered to show only birds from the Genus Corvus observed in Massachusetts after 2014-01-01</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb21"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb21-1"><a href="#cb21-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load complete bird data</span></span>
<span id="cb21-2"><a href="#cb21-2" aria-hidden="true" tabindex="-1"></a>bird <span class="ot">&lt;-</span> <span class="fu">read.table</span>(<span class="st">"/Users/andrewruiz/0011508-240216155721649.csv"</span>, <span class="at">sep =</span> <span class="st">"</span><span class="sc">\t</span><span class="st">"</span>, <span class="at">header =</span> <span class="cn">TRUE</span>, <span class="at">fill =</span> <span class="cn">TRUE</span>)</span>
<span id="cb21-3"><a href="#cb21-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-4"><a href="#cb21-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Select variables to keep and then filter records based on order, stateProvince, and eventDate</span></span>
<span id="cb21-5"><a href="#cb21-5" aria-hidden="true" tabindex="-1"></a>bird_select <span class="ot">&lt;-</span> bird <span class="sc">%&gt;%</span></span>
<span id="cb21-6"><a href="#cb21-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(genus, </span>
<span id="cb21-7"><a href="#cb21-7" aria-hidden="true" tabindex="-1"></a>         species,</span>
<span id="cb21-8"><a href="#cb21-8" aria-hidden="true" tabindex="-1"></a>         stateProvince, </span>
<span id="cb21-9"><a href="#cb21-9" aria-hidden="true" tabindex="-1"></a>         decimalLatitude, </span>
<span id="cb21-10"><a href="#cb21-10" aria-hidden="true" tabindex="-1"></a>         decimalLongitude, </span>
<span id="cb21-11"><a href="#cb21-11" aria-hidden="true" tabindex="-1"></a>         eventDate) <span class="sc">%&gt;%</span></span>
<span id="cb21-12"><a href="#cb21-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">eventDate =</span> <span class="fu">as.Date</span>(eventDate, <span class="at">format =</span> <span class="st">"%Y-%m-%d"</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb21-13"><a href="#cb21-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(eventDate <span class="sc">&gt;</span> <span class="fu">as.Date</span>(<span class="st">"2014-01-01"</span>), genus <span class="sc">==</span> <span class="st">"Corvus"</span>, stateProvince <span class="sc">==</span> <span class="st">"Massachusetts"</span>)</span>
<span id="cb21-14"><a href="#cb21-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb21-15"><a href="#cb21-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Inspect the structure of the filtered dataset</span></span>
<span id="cb21-16"><a href="#cb21-16" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(bird_select)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>'data.frame':   12293 obs. of  6 variables:
 $ genus           : chr  "Corvus" "Corvus" "Corvus" "Corvus" ...
 $ species         : chr  "Corvus brachyrhynchos" "Corvus brachyrhynchos" "Corvus brachyrhynchos" "Corvus brachyrhynchos" ...
 $ stateProvince   : chr  "Massachusetts" "Massachusetts" "Massachusetts" "Massachusetts" ...
 $ decimalLatitude : chr  "41.720417" "41.50707" "41.945" "41.737503" ...
 $ decimalLongitude: chr  "-70.75801" "-71.02362" "-70.879" "-70.84186" ...
 $ eventDate       : Date, format: "2021-09-18" "2020-07-10" ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb23"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb23-1"><a href="#cb23-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the file path using here()</span></span>
<span id="cb23-2"><a href="#cb23-2" aria-hidden="true" tabindex="-1"></a>output_file_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"filtered_bird_data.csv"</span>)</span>
<span id="cb23-3"><a href="#cb23-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-4"><a href="#cb23-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the bird_select dataframe to the specified folder</span></span>
<span id="cb23-5"><a href="#cb23-5" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(bird_select, output_file_path, <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-6"><a href="#cb23-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-7"><a href="#cb23-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the filtered bird data to R</span></span>
<span id="cb23-8"><a href="#cb23-8" aria-hidden="true" tabindex="-1"></a>filtered_bird <span class="ot">=</span> <span class="fu">read_csv</span>(output_file_path, <span class="at">show_col_types =</span> <span class="cn">FALSE</span>)</span>
<span id="cb23-9"><a href="#cb23-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-10"><a href="#cb23-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the combined dataset RDS</span></span>
<span id="cb23-11"><a href="#cb23-11" aria-hidden="true" tabindex="-1"></a>bird_path_rds <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"filtered_bird.rds"</span>)</span>
<span id="cb23-12"><a href="#cb23-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-13"><a href="#cb23-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the bird data frame as an RDS file</span></span>
<span id="cb23-14"><a href="#cb23-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(filtered_bird, bird_path_rds)</span>
<span id="cb23-15"><a href="#cb23-15" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb23-16"><a href="#cb23-16" aria-hidden="true" tabindex="-1"></a><span class="co"># ensure the csv loaded properly</span></span>
<span id="cb23-17"><a href="#cb23-17" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(filtered_bird)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>spc_tbl_ [12,293 × 6] (S3: spec_tbl_df/tbl_df/tbl/data.frame)
 $ genus           : chr [1:12293] "Corvus" "Corvus" "Corvus" "Corvus" ...
 $ species         : chr [1:12293] "Corvus brachyrhynchos" "Corvus brachyrhynchos" "Corvus brachyrhynchos" "Corvus brachyrhynchos" ...
 $ stateProvince   : chr [1:12293] "Massachusetts" "Massachusetts" "Massachusetts" "Massachusetts" ...
 $ decimalLatitude : num [1:12293] 41.7 41.5 41.9 41.7 41.6 ...
 $ decimalLongitude: num [1:12293] -70.8 -71 -70.9 -70.8 -71 ...
 $ eventDate       : Date[1:12293], format: "2021-09-18" "2020-07-10" ...
 - attr(*, "spec")=
  .. cols(
  ..   genus = col_character(),
  ..   species = col_character(),
  ..   stateProvince = col_character(),
  ..   decimalLatitude = col_double(),
  ..   decimalLongitude = col_double(),
  ..   eventDate = col_date(format = "")
  .. )
 - attr(*, "problems")=&lt;externalptr&gt; </code></pre>
</div>
<div class="sourceCode cell-code" id="cb25"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb25-1"><a href="#cb25-1" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(filtered_bird)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  genus  species       stateProvince decimalLatitude decimalLongitude eventDate 
  &lt;chr&gt;  &lt;chr&gt;         &lt;chr&gt;                   &lt;dbl&gt;            &lt;dbl&gt; &lt;date&gt;    
1 Corvus Corvus brach… Massachusetts            41.7            -70.8 2021-09-18
2 Corvus Corvus brach… Massachusetts            41.5            -71.0 2020-07-10
3 Corvus Corvus brach… Massachusetts            41.9            -70.9 2018-09-29
4 Corvus Corvus brach… Massachusetts            41.7            -70.8 2014-11-15
5 Corvus Corvus brach… Massachusetts            41.6            -71.0 2022-09-13
6 Corvus Corvus brach… Massachusetts            41.7            -70.9 2019-07-24</code></pre>
</div>
</div>
</section>
</section>
<section id="weather-data" class="level2">
<h2 class="anchored" data-anchor-id="weather-data">Weather data</h2>
<section id="this-dataset-contains-daily-weather-summaries-from-the-taunton-ma-airport-from-2007-2023" class="level3">
<h3 class="anchored" data-anchor-id="this-dataset-contains-daily-weather-summaries-from-the-taunton-ma-airport-from-2007-2023">This dataset contains daily weather summaries from the Taunton, MA Airport from 2007-2023</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb27"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb27-1"><a href="#cb27-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct the path to the file</span></span>
<span id="cb27-2"><a href="#cb27-2" aria-hidden="true" tabindex="-1"></a>file_path_wx <span class="ot">=</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"weather_airport.csv"</span>)</span>
<span id="cb27-3"><a href="#cb27-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-4"><a href="#cb27-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import the data to R and store it in a dataframe named "wx"</span></span>
<span id="cb27-5"><a href="#cb27-5" aria-hidden="true" tabindex="-1"></a>wx <span class="ot">=</span> <span class="fu">read.csv</span>(file_path_wx)</span>
<span id="cb27-6"><a href="#cb27-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb27-7"><a href="#cb27-7" aria-hidden="true" tabindex="-1"></a><span class="co">#load the first few rows of data to verify the data loaded as expected</span></span>
<span id="cb27-8"><a href="#cb27-8" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wx)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      STATION                             NAME   DATE  AWND AWND_ATTRIBUTES
1 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/1/07  6.49             ,,W
2 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/2/07  8.95             ,,W
3 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/3/07  7.61             ,,W
4 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/4/07  6.71             ,,W
5 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/5/07  8.28             ,,W
6 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/6/07 10.51             ,,W
  PRCP PRCP_ATTRIBUTES TMAX TMAX_ATTRIBUTES TMIN TMIN_ATTRIBUTES WT01
1 1.17             ,,W   57             ,,W   26             ,,W    1
2 0.01             ,,W   48             ,,W   34             ,,W   NA
3 0.00             ,,W   50             ,,W   25             ,,W   NA
4 0.00             ,,W   54             ,,W   37             ,,W   NA
5 0.04             ,,W   60             ,,W   48             ,,W    1
6 0.05             ,,W   63             ,,W   54             ,,W    1
  WT01_ATTRIBUTES WT02 WT02_ATTRIBUTES WT03 WT03_ATTRIBUTES WT08
1             ,,W   NA                   NA                   NA
2                   NA                   NA                   NA
3                   NA                   NA                   NA
4                   NA                   NA                   NA
5             ,,W   NA                   NA                   NA
6             ,,W   NA                   NA                   NA
  WT08_ATTRIBUTES
1                
2                
3                
4                
5                
6                </code></pre>
</div>
<div class="sourceCode cell-code" id="cb29"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb29-1"><a href="#cb29-1" aria-hidden="true" tabindex="-1"></a><span class="co">#the columns below are the only ones needed for this project. </span></span>
<span id="cb29-2"><a href="#cb29-2" aria-hidden="true" tabindex="-1"></a>wx_selected <span class="ot">&lt;-</span> wx <span class="sc">%&gt;%</span></span>
<span id="cb29-3"><a href="#cb29-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">select</span>(STATION, NAME, DATE, PRCP)</span>
<span id="cb29-4"><a href="#cb29-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-5"><a href="#cb29-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the RDS file</span></span>
<span id="cb29-6"><a href="#cb29-6" aria-hidden="true" tabindex="-1"></a>output_path_wx <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"wx_selected.rds"</span>)</span>
<span id="cb29-7"><a href="#cb29-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-8"><a href="#cb29-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the wx_selected dataframe as an RDS file</span></span>
<span id="cb29-9"><a href="#cb29-9" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(wx_selected, output_path_wx)</span>
<span id="cb29-10"><a href="#cb29-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb29-11"><a href="#cb29-11" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wx_selected)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>      STATION                             NAME   DATE PRCP
1 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/1/07 1.17
2 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/2/07 0.01
3 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/3/07 0.00
4 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/4/07 0.00
5 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/5/07 0.04
6 USW00054777 TAUNTON MUNICIPAL AIRPORT, MA US 1/6/07 0.05</code></pre>
</div>
</div>
</section>
</section>
<section id="human-and-veterinary-cases-of-eastern-equine-encephalitis-and-west-nile-virus" class="level2">
<h2 class="anchored" data-anchor-id="human-and-veterinary-cases-of-eastern-equine-encephalitis-and-west-nile-virus">Human and veterinary cases of eastern equine encephalitis and West Nile virus</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb31"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb31-1"><a href="#cb31-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct the path to the human case data file</span></span>
<span id="cb31-2"><a href="#cb31-2" aria-hidden="true" tabindex="-1"></a>file_path_human <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"human_cases.xlsx"</span>)</span>
<span id="cb31-3"><a href="#cb31-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-4"><a href="#cb31-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import the data to R and store it in a dataframe named "cases_human"</span></span>
<span id="cb31-5"><a href="#cb31-5" aria-hidden="true" tabindex="-1"></a>cases_human <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path_human)</span>
<span id="cb31-6"><a href="#cb31-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-7"><a href="#cb31-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'cases_human' as an RDS file</span></span>
<span id="cb31-8"><a href="#cb31-8" aria-hidden="true" tabindex="-1"></a>output_path_rds_human <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"human_cases_processed.rds"</span>)</span>
<span id="cb31-9"><a href="#cb31-9" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cases_human, output_path_rds_human)</span>
<span id="cb31-10"><a href="#cb31-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb31-11"><a href="#cb31-11" aria-hidden="true" tabindex="-1"></a><span class="co">#load the first few rows of data to verify the data loaded as expected</span></span>
<span id="cb31-12"><a href="#cb31-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cases_human)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 5
  County   `Age Range` `Onset Date`        `Virus Result` Clinical Presentatio…¹
  &lt;chr&gt;    &lt;chr&gt;       &lt;dttm&gt;              &lt;chr&gt;          &lt;chr&gt;                 
1 Middles… 61-70       2014-08-11 00:00:00 WNV            meningoencephalitis   
2 Middles… 41-50       2014-08-22 00:00:00 WNV            meningoencephalitis   
3 Essex    51-60       2014-08-30 00:00:00 WNV            meningitis            
4 Middles… 21-30       2014-09-05 00:00:00 WNV            fever                 
5 Middles… 41-50       2014-09-15 00:00:00 WNV            meningitis            
6 Middles… 41-50       2014-09-28 00:00:00 WNV            meningoencephalitis   
# ℹ abbreviated name: ¹​`Clinical Presentation`</code></pre>
</div>
<div class="sourceCode cell-code" id="cb33"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb33-1"><a href="#cb33-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Construct the path to the veterinary case data file</span></span>
<span id="cb33-2"><a href="#cb33-2" aria-hidden="true" tabindex="-1"></a>file_path_vet <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"animal_cases.xlsx"</span>)</span>
<span id="cb33-3"><a href="#cb33-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-4"><a href="#cb33-4" aria-hidden="true" tabindex="-1"></a><span class="co"># import the data to R and store it in a dataframe named "cases_human"</span></span>
<span id="cb33-5"><a href="#cb33-5" aria-hidden="true" tabindex="-1"></a>cases_vet <span class="ot">&lt;-</span> <span class="fu">read_excel</span>(file_path_vet)</span>
<span id="cb33-6"><a href="#cb33-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-7"><a href="#cb33-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Save 'cases_vet' as an RDS file</span></span>
<span id="cb33-8"><a href="#cb33-8" aria-hidden="true" tabindex="-1"></a>output_path_rds_vet <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"vet_cases_processed.rds"</span>)</span>
<span id="cb33-9"><a href="#cb33-9" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(cases_vet, output_path_rds_vet)</span>
<span id="cb33-10"><a href="#cb33-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb33-11"><a href="#cb33-11" aria-hidden="true" tabindex="-1"></a><span class="co">#load the first few rows of data to verify the data loaded as expected</span></span>
<span id="cb33-12"><a href="#cb33-12" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(cases_vet)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 6
  date_onset           year animal city            county    virus
  &lt;dttm&gt;              &lt;dbl&gt; &lt;chr&gt;  &lt;chr&gt;           &lt;chr&gt;     &lt;chr&gt;
1 2014-08-09 00:00:00  2014 deer   Freetown        Bristol   eee  
2 2014-10-05 00:00:00  2014 horse  Westminster     Worcester eee  
3 NA                   2018 turkey Hubbardston     Worcester eee  
4 NA                   2018 horse  East Brookfield Worcester eee  
5 NA                   2018 horse  Bernardston     Franklin  wnv  
6 NA                   2018 horse  Granby          Hampshire wnv  </code></pre>
</div>
<div class="sourceCode cell-code" id="cb35"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb35-1"><a href="#cb35-1" aria-hidden="true" tabindex="-1"></a><span class="co">#date of onset is missing for some records, however, they are not needed for this project. </span></span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="processing-and-cleaning-the-mosquito-trap-data" class="level1">
<h1>Processing and cleaning the Mosquito trap data</h1>
<section id="examine-the-variables-and-change-all-column-names-to-lower-case" class="level3">
<h3 class="anchored" data-anchor-id="examine-the-variables-and-change-all-column-names-to-lower-case">Examine the variables and change all column names to lower case</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb36"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb36-1"><a href="#cb36-1" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(mosquito_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [45,389 × 12] (S3: tbl_df/tbl/data.frame)
 $ MCD                  : chr [1:45389] "Bristol" "State Lab Institute" "Bristol" "Bristol" ...
 $ Pool ID              : chr [1:45389] "BR21NS-0575" "SL21NS-6080" "BR22NS-0462" "BR07-00013" ...
 $ Species code         : chr [1:45389] "PRO" "TVT" "WAL" "ABS" ...
 $ Trap Set Date        : POSIXct[1:45389], format: "2021-07-26" "2021-08-29" ...
 $ Collection Date      : POSIXct[1:45389], format: "2021-07-27" "2021-08-30" ...
 $ Town                 : chr [1:45389] "Berkley" "Raynham" "Mansfield" "Freetown" ...
 $ Number of Traps      : num [1:45389] 1 1 1 1 1 1 1 1 1 1 ...
 $ Trap Type            : chr [1:45389] "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" ...
 $ Pool Size            : num [1:45389] 3 1 1 50 44 50 50 50 50 50 ...
 $ Submitted for Testing: chr [1:45389] "N" "N" "N" "Y" ...
 $ Genus                : chr [1:45389] "Ochlerotatus" "Ochlerotatus" "Anopheles" "Ochlerotatus" ...
 $ Species              : chr [1:45389] "provocans" "trivittatus" "walkeri" "abserratus" ...</code></pre>
</div>
<div class="sourceCode cell-code" id="cb38"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb38-1"><a href="#cb38-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Set column names to lowercase</span></span>
<span id="cb38-2"><a href="#cb38-2" aria-hidden="true" tabindex="-1"></a><span class="fu">names</span>(mosquito_raw) <span class="ot">&lt;-</span> <span class="fu">tolower</span>(<span class="fu">names</span>(mosquito_raw))</span>
<span id="cb38-3"><a href="#cb38-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb38-4"><a href="#cb38-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the first few rows to confirm changes</span></span>
<span id="cb38-5"><a href="#cb38-5" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(mosquito_raw)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 12
  mcd     `pool id` `species code` `trap set date`     `collection date`   town 
  &lt;chr&gt;   &lt;chr&gt;     &lt;chr&gt;          &lt;dttm&gt;              &lt;dttm&gt;              &lt;chr&gt;
1 Bristol BR21NS-0… PRO            2021-07-26 00:00:00 2021-07-27 00:00:00 Berk…
2 State … SL21NS-6… TVT            2021-08-29 00:00:00 2021-08-30 00:00:00 Rayn…
3 Bristol BR22NS-0… WAL            2022-07-07 00:00:00 2022-07-08 00:00:00 Mans…
4 Bristol BR07-000… ABS            2007-06-04 00:00:00 2007-06-05 00:00:00 Free…
5 Bristol BR07-000… ABS            2007-06-04 00:00:00 2007-06-05 00:00:00 Free…
6 Bristol BR07-000… ABS            2007-06-06 00:00:00 2007-06-07 00:00:00 Free…
# ℹ 6 more variables: `number of traps` &lt;dbl&gt;, `trap type` &lt;chr&gt;,
#   `pool size` &lt;dbl&gt;, `submitted for testing` &lt;chr&gt;, genus &lt;chr&gt;,
#   species &lt;chr&gt;</code></pre>
</div>
</div>
</section>
<section id="identify-the-time-period-the-dataset-covers" class="level3">
<h3 class="anchored" data-anchor-id="identify-the-time-period-the-dataset-covers">Identify the time period the dataset covers</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb40"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb40-1"><a href="#cb40-1" aria-hidden="true" tabindex="-1"></a>earliest_date <span class="ot">&lt;-</span> <span class="fu">min</span>(mosquito_raw<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-2"><a href="#cb40-2" aria-hidden="true" tabindex="-1"></a>latest_date <span class="ot">&lt;-</span> <span class="fu">max</span>(mosquito_raw<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>)</span>
<span id="cb40-3"><a href="#cb40-3" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"Date range:"</span>, earliest_date,<span class="st">" to "</span>, latest_date))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "Date range: 2007-05-10  to  2023-11-01"</code></pre>
</div>
</div>
<section id="the-mosquito-dataset-spans-2007-to-2023.-however-the-virus-and-case-data-only-covers-the-years-2014-to-2020.-because-mosquito-populations-can-be-influences-by-processes-of-the-previous-year.-we-will-set-the-analytical-horizon-will-span-2013-to-2021.collection-data-before-2013-and-after-2021-will-be-removed-along-with-records-with-important-missing-variables-identified-below." class="level4">
<h4 class="anchored" data-anchor-id="the-mosquito-dataset-spans-2007-to-2023.-however-the-virus-and-case-data-only-covers-the-years-2014-to-2020.-because-mosquito-populations-can-be-influences-by-processes-of-the-previous-year.-we-will-set-the-analytical-horizon-will-span-2013-to-2021.collection-data-before-2013-and-after-2021-will-be-removed-along-with-records-with-important-missing-variables-identified-below.">The mosquito dataset spans 2007 to 2023. However, the virus and case data only covers the years 2014 to 2020. Because mosquito populations can be influences by processes of the previous year. We will set the analytical horizon will span 2013 to 2021.<em>Collection data before 2013 and after 2021 will be removed</em> along with records with important missing variables identified below.</h4>
</section>
</section>
<section id="identify-missing-data" class="level3">
<h3 class="anchored" data-anchor-id="identify-missing-data">Identify missing data</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb42"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb42-1"><a href="#cb42-1" aria-hidden="true" tabindex="-1"></a><span class="fu">sapply</span>(mosquito_raw, <span class="cf">function</span>(x) <span class="fu">sum</span>(<span class="fu">is.na</span>(x)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>                  mcd               pool id          species code 
                    0                     0                     6 
        trap set date       collection date                  town 
                    0                     0                     2 
      number of traps             trap type             pool size 
                    0                     0                     9 
submitted for testing                 genus               species 
                   57                     7                     0 </code></pre>
</div>
</div>
<section id="the-majority-of-the-missing-data-is-in-the-submitted-for-testing-variable.-this-is-not-a-crucial-variable-so-we-can-keep-those-records.the-other-variable-are-important-and-so-we-will-remove-those-records-with-the-other-missing-data." class="level4">
<h4 class="anchored" data-anchor-id="the-majority-of-the-missing-data-is-in-the-submitted-for-testing-variable.-this-is-not-a-crucial-variable-so-we-can-keep-those-records.the-other-variable-are-important-and-so-we-will-remove-those-records-with-the-other-missing-data.">The majority of the missing data is in the “Submitted for Testing” variable. This is not a crucial variable, so we can keep those records.The other variable are important and so we will remove those records with the other missing data.</h4>
</section>
</section>
<section id="the-massachussetts-department-of-public-health-adpoted-a-taxonomy-change-before-it-was-fully-vetted.-the-change-divided-the-genus-aedes-into-a-new-genus-ochlerotatus.-to-align-with-accepted-standards-i-will-change-them-back-to-aedes." class="level3">
<h3 class="anchored" data-anchor-id="the-massachussetts-department-of-public-health-adpoted-a-taxonomy-change-before-it-was-fully-vetted.-the-change-divided-the-genus-aedes-into-a-new-genus-ochlerotatus.-to-align-with-accepted-standards-i-will-change-them-back-to-aedes.">The Massachussetts Department of Public Health adpoted a taxonomy change before it was fully vetted. The change divided the genus <em>Aedes</em> into a new genus: <em>Ochlerotatus</em>. To align with accepted standards, I will change them back to <em>Aedes</em>.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb44"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb44-1"><a href="#cb44-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Get unique values using dplyr</span></span>
<span id="cb44-2"><a href="#cb44-2" aria-hidden="true" tabindex="-1"></a>unique_genus_values <span class="ot">&lt;-</span> mosquito_raw <span class="sc">%&gt;%</span></span>
<span id="cb44-3"><a href="#cb44-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">distinct</span>(genus) <span class="sc">%&gt;%</span></span>
<span id="cb44-4"><a href="#cb44-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pull</span>(genus)</span>
<span id="cb44-5"><a href="#cb44-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb44-6"><a href="#cb44-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the unique genus values</span></span>
<span id="cb44-7"><a href="#cb44-7" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(unique_genus_values)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Ochlerotatus"   "Anopheles"      "Aedes"          "Psorophora"    
 [5] "Culex"          "Culiseta"       "Coquillettidia" "Uranotaenia"   
 [9] "No Collections" NA               "no collections" "No collections"
[13] "Toxorhynchites" "Orthopodomyia"  "Wyeomia"        "Unknown"       </code></pre>
</div>
<div class="sourceCode cell-code" id="cb46"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb46-1"><a href="#cb46-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Change records where genus is "Ochlerotatus" to "Aedes"</span></span>
<span id="cb46-2"><a href="#cb46-2" aria-hidden="true" tabindex="-1"></a>mosquito_raw<span class="sc">$</span>genus[mosquito_raw<span class="sc">$</span>genus <span class="sc">==</span> <span class="st">"Ochlerotatus"</span>] <span class="ot">&lt;-</span> <span class="st">"Aedes"</span></span>
<span id="cb46-3"><a href="#cb46-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb46-4"><a href="#cb46-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the first few rows to confirm changes, or use unique() to see the change in genus values</span></span>
<span id="cb46-5"><a href="#cb46-5" aria-hidden="true" tabindex="-1"></a><span class="fu">unique</span>(mosquito_raw<span class="sc">$</span>genus)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code> [1] "Aedes"          "Anopheles"      "Psorophora"     "Culex"         
 [5] "Culiseta"       "Coquillettidia" "Uranotaenia"    "No Collections"
 [9] NA               "no collections" "No collections" "Toxorhynchites"
[13] "Orthopodomyia"  "Wyeomia"        "Unknown"       </code></pre>
</div>
</div>
</section>
<section id="execute-proposed-changes." class="level3">
<h3 class="anchored" data-anchor-id="execute-proposed-changes.">Execute proposed changes.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb48"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb48-1"><a href="#cb48-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create a copy of the dataframe to avoid modifying the original</span></span>
<span id="cb48-2"><a href="#cb48-2" aria-hidden="true" tabindex="-1"></a>mosquito_modified <span class="ot">&lt;-</span> mosquito_raw</span>
<span id="cb48-3"><a href="#cb48-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-4"><a href="#cb48-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Change records where genus is "Ochlerotatus" to "Aedes"</span></span>
<span id="cb48-5"><a href="#cb48-5" aria-hidden="true" tabindex="-1"></a><span class="co">#mosquito_modified$genus[mosquito_modified$genus == "Ochlerotatus"] &lt;- "Aedes"</span></span>
<span id="cb48-6"><a href="#cb48-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-7"><a href="#cb48-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Filter out records where Collection Date is before 2013 or after 2021</span></span>
<span id="cb48-8"><a href="#cb48-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure Collection Date is in Date format if it's not already</span></span>
<span id="cb48-9"><a href="#cb48-9" aria-hidden="true" tabindex="-1"></a>mosquito_modified<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span> <span class="ot">&lt;-</span> <span class="fu">as.Date</span>(mosquito_modified<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span>)</span>
<span id="cb48-10"><a href="#cb48-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-11"><a href="#cb48-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Convert all species code values to uppercase</span></span>
<span id="cb48-12"><a href="#cb48-12" aria-hidden="true" tabindex="-1"></a>mosquito_modified <span class="ot">&lt;-</span> mosquito_modified <span class="sc">%&gt;%</span></span>
<span id="cb48-13"><a href="#cb48-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="st">`</span><span class="at">species code</span><span class="st">`</span> <span class="ot">=</span> <span class="fu">toupper</span>(<span class="st">`</span><span class="at">species code</span><span class="st">`</span>))</span>
<span id="cb48-14"><a href="#cb48-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-15"><a href="#cb48-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Now apply the date filter</span></span>
<span id="cb48-16"><a href="#cb48-16" aria-hidden="true" tabindex="-1"></a>mosquito_modified <span class="ot">&lt;-</span> mosquito_modified[mosquito_modified<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span> <span class="sc">&gt;=</span> <span class="fu">as.Date</span>(<span class="st">"2013-01-01"</span>) <span class="sc">&amp;</span> </span>
<span id="cb48-17"><a href="#cb48-17" aria-hidden="true" tabindex="-1"></a>                                       mosquito_modified<span class="sc">$</span><span class="st">`</span><span class="at">collection date</span><span class="st">`</span> <span class="sc">&lt;=</span> <span class="fu">as.Date</span>(<span class="st">"2021-12-31"</span>), ]</span>
<span id="cb48-18"><a href="#cb48-18" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-19"><a href="#cb48-19" aria-hidden="true" tabindex="-1"></a><span class="co"># Remove rows with NA values in the remaining columns of the modified dataframe, </span></span>
<span id="cb48-20"><a href="#cb48-20" aria-hidden="true" tabindex="-1"></a><span class="co"># except for "Submitted for Testing" which we'll handle separately</span></span>
<span id="cb48-21"><a href="#cb48-21" aria-hidden="true" tabindex="-1"></a>temp_columns <span class="ot">&lt;-</span> <span class="fu">names</span>(mosquito_modified)</span>
<span id="cb48-22"><a href="#cb48-22" aria-hidden="true" tabindex="-1"></a>temp_columns <span class="ot">&lt;-</span> temp_columns[temp_columns <span class="sc">!=</span> <span class="st">"submitted for testing"</span>]</span>
<span id="cb48-23"><a href="#cb48-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-24"><a href="#cb48-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Applying NA omit on selected columns only</span></span>
<span id="cb48-25"><a href="#cb48-25" aria-hidden="true" tabindex="-1"></a>mosquito_clean_except_submitted <span class="ot">&lt;-</span> <span class="fu">na.omit</span>(mosquito_modified[temp_columns])</span>
<span id="cb48-26"><a href="#cb48-26" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-27"><a href="#cb48-27" aria-hidden="true" tabindex="-1"></a><span class="co"># Since na.omit on selected columns messes with row alignment, let's filter the original dataset based on the cleaned IDs</span></span>
<span id="cb48-28"><a href="#cb48-28" aria-hidden="true" tabindex="-1"></a>clean_ids <span class="ot">&lt;-</span> <span class="fu">row.names</span>(mosquito_clean_except_submitted)</span>
<span id="cb48-29"><a href="#cb48-29" aria-hidden="true" tabindex="-1"></a>mosquito_final <span class="ot">&lt;-</span> mosquito_modified[clean_ids, ]</span>
<span id="cb48-30"><a href="#cb48-30" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-31"><a href="#cb48-31" aria-hidden="true" tabindex="-1"></a><span class="co"># Add back "Submitted for Testing" from the original dataframe if needed</span></span>
<span id="cb48-32"><a href="#cb48-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: This step might be redundant if "Submitted for Testing" was not actually removed or if it's included in the cleaning process</span></span>
<span id="cb48-33"><a href="#cb48-33" aria-hidden="true" tabindex="-1"></a>mosquito_final<span class="sc">$</span><span class="st">`</span><span class="at">submitted for testing</span><span class="st">`</span> <span class="ot">&lt;-</span> mosquito_raw<span class="sc">$</span><span class="st">`</span><span class="at">submitted for testing</span><span class="st">`</span>[clean_ids]</span>
<span id="cb48-34"><a href="#cb48-34" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb48-35"><a href="#cb48-35" aria-hidden="true" tabindex="-1"></a><span class="co"># Check the structure of the final cleaned dataframe</span></span>
<span id="cb48-36"><a href="#cb48-36" aria-hidden="true" tabindex="-1"></a><span class="fu">str</span>(mosquito_final)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>tibble [23,995 × 12] (S3: tbl_df/tbl/data.frame)
 $ mcd                  : chr [1:23995] "Bristol" "State Lab Institute" "Bristol" "Bristol" ...
 $ pool id              : chr [1:23995] "BR21NS-0575" "SL21NS-6080" "BR13NS-0002" "BR13NS-0008" ...
 $ species code         : chr [1:23995] "PRO" "TVT" "ABS" "ABS" ...
 $ trap set date        : POSIXct[1:23995], format: "2021-07-26" "2021-08-29" ...
 $ collection date      : Date[1:23995], format: "2021-07-27" "2021-08-30" ...
 $ town                 : chr [1:23995] "Berkley" "Raynham" "Easton" "Taunton" ...
 $ number of traps      : num [1:23995] 1 1 1 1 1 1 1 1 1 1 ...
 $ trap type            : chr [1:23995] "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" "500cc CO2 - CDC Miniature Light Trap" ...
 $ pool size            : num [1:23995] 3 1 59 41 54 5 4 6 5 6 ...
 $ submitted for testing: chr [1:23995] NA NA NA NA ...
 $ genus                : chr [1:23995] "Aedes" "Aedes" "Aedes" "Aedes" ...
 $ species              : chr [1:23995] "provocans" "trivittatus" "abserratus" "abserratus" ...</code></pre>
</div>
</div>
</section>
<section id="reformat-the-collection-data" class="level2">
<h2 class="anchored" data-anchor-id="reformat-the-collection-data">Reformat the collection data</h2>
<section id="the-massachusetts-department-of-public-health-requires-that-all-mosquito-control-district-submit-their-colelction-data-in-a-specific-format.-this-format-divides-each-trap-event-into-separate-rows-based-on-mosquito-species.-however-this-format-is-not-useful-for-certain-calculations-where-a-zero-count-for-a-species-in-not-explicitly-recorded.-even-if-you-sort-by-species-and-calculate-average-count-per-trap-the-average-could-be-inflated-since-the-records-will-not-include-trap-events-with-a-zero-count-of-that-species." class="level3">
<h3 class="anchored" data-anchor-id="the-massachusetts-department-of-public-health-requires-that-all-mosquito-control-district-submit-their-colelction-data-in-a-specific-format.-this-format-divides-each-trap-event-into-separate-rows-based-on-mosquito-species.-however-this-format-is-not-useful-for-certain-calculations-where-a-zero-count-for-a-species-in-not-explicitly-recorded.-even-if-you-sort-by-species-and-calculate-average-count-per-trap-the-average-could-be-inflated-since-the-records-will-not-include-trap-events-with-a-zero-count-of-that-species.">The Massachusetts Department of Public Health requires that all mosquito control district submit their colelction data in a specific format. This format divides each trap event into separate rows based on mosquito species. However, this format is not useful for certain calculations where a zero count for a species in not explicitly recorded. Even if you sort by species and calculate average count per trap, the average could be inflated since the records will not include trap events with a zero count of that species.</h3>
</section>
<section id="in-order-to-correct-this-i-will-pivot-the-data-and-create-a-column-for-every-unique-species-code-and-sum-the-pool-size-for-each-species-code.-the-resulting-table-will-have-one-row-for-every-trap-event--a-trap-event-is-when-town-date-of-collection-trap-type-are-all-the-same." class="level3">
<h3 class="anchored" data-anchor-id="in-order-to-correct-this-i-will-pivot-the-data-and-create-a-column-for-every-unique-species-code-and-sum-the-pool-size-for-each-species-code.-the-resulting-table-will-have-one-row-for-every-trap-event--a-trap-event-is-when-town-date-of-collection-trap-type-are-all-the-same.">In order to correct this, I will pivot the data and create a column for every unique “species code” and sum the “pool size” for each species code. The resulting table will have one row for every trap event -a trap event is when “town”, “date of collection”, “trap type” are all the same.</h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb50"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb50-1"><a href="#cb50-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb50-2"><a href="#cb50-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(tidyr)</span>
<span id="cb50-3"><a href="#cb50-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-4"><a href="#cb50-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Group and summarize the mosquito data to prepare for wide format transformation.</span></span>
<span id="cb50-5"><a href="#cb50-5" aria-hidden="true" tabindex="-1"></a><span class="co"># This step aggregates the total 'pool size' for each unique combination of </span></span>
<span id="cb50-6"><a href="#cb50-6" aria-hidden="true" tabindex="-1"></a><span class="co"># 'collection date', 'town', 'trap type', 'species code', and 'mcd' (Mosquito Control District),</span></span>
<span id="cb50-7"><a href="#cb50-7" aria-hidden="true" tabindex="-1"></a><span class="co"># while excluding 'submitted for testing' from the grouping to avoid potential inconsistencies.</span></span>
<span id="cb50-8"><a href="#cb50-8" aria-hidden="true" tabindex="-1"></a>mosquito_summarized <span class="ot">&lt;-</span> mosquito_final <span class="sc">%&gt;%</span></span>
<span id="cb50-9"><a href="#cb50-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(<span class="st">`</span><span class="at">collection date</span><span class="st">`</span>, town, <span class="st">`</span><span class="at">trap type</span><span class="st">`</span>, <span class="st">`</span><span class="at">species code</span><span class="st">`</span>, mcd, <span class="st">`</span><span class="at">submitted for testing</span><span class="st">`</span>) <span class="sc">%&gt;%</span></span>
<span id="cb50-10"><a href="#cb50-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">pool_size_sum =</span> <span class="fu">sum</span>(<span class="st">`</span><span class="at">pool size</span><span class="st">`</span>, <span class="at">na.rm =</span> <span class="cn">TRUE</span>), <span class="at">.groups =</span> <span class="st">'drop'</span>)</span>
<span id="cb50-11"><a href="#cb50-11" aria-hidden="true" tabindex="-1"></a>   <span class="co"># Ensure that the summarization does not drop any groups by setting .groups = 'drop'.</span></span>
<span id="cb50-12"><a href="#cb50-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-13"><a href="#cb50-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-14"><a href="#cb50-14" aria-hidden="true" tabindex="-1"></a><span class="co"># Transform the summarized mosquito data to a wide format where each 'species code' becomes a separate column.</span></span>
<span id="cb50-15"><a href="#cb50-15" aria-hidden="true" tabindex="-1"></a><span class="co"># This pivot operation facilitates easier analysis of mosquito counts by species across different events.</span></span>
<span id="cb50-16"><a href="#cb50-16" aria-hidden="true" tabindex="-1"></a><span class="co"># 'values_fill' is used to fill in zeroes for any 'species code' not caught in a specific trapping event,</span></span>
<span id="cb50-17"><a href="#cb50-17" aria-hidden="true" tabindex="-1"></a><span class="co"># ensuring that the dataset has a consistent shape and making it more analyzable.</span></span>
<span id="cb50-18"><a href="#cb50-18" aria-hidden="true" tabindex="-1"></a>wide_data <span class="ot">&lt;-</span> mosquito_summarized <span class="sc">%&gt;%</span></span>
<span id="cb50-19"><a href="#cb50-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">pivot_wider</span>(</span>
<span id="cb50-20"><a href="#cb50-20" aria-hidden="true" tabindex="-1"></a>    <span class="at">names_from =</span> <span class="st">`</span><span class="at">species code</span><span class="st">`</span>,</span>
<span id="cb50-21"><a href="#cb50-21" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_from =</span> pool_size_sum,</span>
<span id="cb50-22"><a href="#cb50-22" aria-hidden="true" tabindex="-1"></a>    <span class="at">values_fill =</span> <span class="fu">list</span>(<span class="at">pool_size_sum =</span> <span class="dv">0</span>),</span>
<span id="cb50-23"><a href="#cb50-23" aria-hidden="true" tabindex="-1"></a>    <span class="at">id_cols =</span> <span class="fu">c</span>(<span class="st">`</span><span class="at">collection date</span><span class="st">`</span>, town, <span class="st">`</span><span class="at">trap type</span><span class="st">`</span>, mcd)</span>
<span id="cb50-24"><a href="#cb50-24" aria-hidden="true" tabindex="-1"></a>  )</span>
<span id="cb50-25"><a href="#cb50-25" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb50-26"><a href="#cb50-26" aria-hidden="true" tabindex="-1"></a><span class="co"># Output the transformed wide-format data to inspect its structure and content.</span></span>
<span id="cb50-27"><a href="#cb50-27" aria-hidden="true" tabindex="-1"></a><span class="co"># This step is crucial for verifying that the pivot operation produced the expected wide-format dataset,</span></span>
<span id="cb50-28"><a href="#cb50-28" aria-hidden="true" tabindex="-1"></a><span class="co"># with each row representing a unique trapping event and columns for each mosquito species.</span></span>
<span id="cb50-29"><a href="#cb50-29" aria-hidden="true" tabindex="-1"></a><span class="fu">head</span>(wide_data)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 6 × 55
  `collection date` town   `trap type` mcd     ABS   CAN   GRO   MEL   PUN   EXC
  &lt;date&gt;            &lt;chr&gt;  &lt;chr&gt;       &lt;chr&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;
1 2013-05-21        Easton 500cc CO2 … Bris…    59     1     2    15     0     0
2 2013-05-21        Raynh… 500cc CO2 … Bris…    54    21     0     8     2     0
3 2013-05-21        Taunt… 500cc CO2 … Bris…    41    59     5     1     0     0
4 2013-05-22        Acush… 500cc CO2 … Bris…     5     2     1     0     0     1
5 2013-05-22        Berkl… 500cc CO2 … Bris…     0     4     0     1     1     0
6 2013-05-22        Freet… 500cc CO2 … Bris…     0     1     0     1     0     0
# ℹ 45 more variables: QUA &lt;dbl&gt;, AUR &lt;dbl&gt;, STI &lt;dbl&gt;, TRI &lt;dbl&gt;, VEX &lt;dbl&gt;,
#   RES &lt;dbl&gt;, JAP &lt;dbl&gt;, PRE &lt;dbl&gt;, CIN &lt;dbl&gt;, PIP &lt;dbl&gt;, NCR &lt;dbl&gt;,
#   PER &lt;dbl&gt;, SAL &lt;dbl&gt;, FER &lt;dbl&gt;, TVT &lt;dbl&gt;, TER &lt;dbl&gt;, FIT &lt;dbl&gt;,
#   ALB &lt;dbl&gt;, CTT &lt;dbl&gt;, SAP &lt;dbl&gt;, SOL &lt;dbl&gt;, MOR &lt;dbl&gt;, ATR &lt;dbl&gt;,
#   TAE &lt;dbl&gt;, CUL &lt;dbl&gt;, WAL &lt;dbl&gt;, ERR &lt;dbl&gt;, CRU &lt;dbl&gt;, MIN &lt;dbl&gt;,
#   HEN &lt;dbl&gt;, RUT &lt;dbl&gt;, IMP &lt;dbl&gt;, INO &lt;dbl&gt;, PRO &lt;dbl&gt;, PCT &lt;dbl&gt;,
#   INT &lt;dbl&gt;, THI &lt;dbl&gt;, SMI &lt;dbl&gt;, STM &lt;dbl&gt;, COM &lt;dbl&gt;, BAR &lt;dbl&gt;, …</code></pre>
</div>
<div class="sourceCode cell-code" id="cb52"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb52-1"><a href="#cb52-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the new file</span></span>
<span id="cb52-2"><a href="#cb52-2" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify the file path using here()</span></span>
<span id="cb52-3"><a href="#cb52-3" aria-hidden="true" tabindex="-1"></a>file_path_wide <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"mosquito_wide.csv"</span>)</span>
<span id="cb52-4"><a href="#cb52-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-5"><a href="#cb52-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the wide-format data to the specified path</span></span>
<span id="cb52-6"><a href="#cb52-6" aria-hidden="true" tabindex="-1"></a><span class="fu">write_csv</span>(wide_data, file_path_wide)</span>
<span id="cb52-7"><a href="#cb52-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb52-8"><a href="#cb52-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Print the file path to confirm where the file has been saved</span></span>
<span id="cb52-9"><a href="#cb52-9" aria-hidden="true" tabindex="-1"></a><span class="fu">print</span>(<span class="fu">paste</span>(<span class="st">"File saved to:"</span>, file_path_wide))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stdout">
<pre><code>[1] "File saved to: /Users/andrewruiz/MADA_course/RUIZ-MADA-project/data/processed-data/mosquito_wide.csv"</code></pre>
</div>
<div class="sourceCode cell-code" id="cb54"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb54-1"><a href="#cb54-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the wide-format dataset RDS</span></span>
<span id="cb54-2"><a href="#cb54-2" aria-hidden="true" tabindex="-1"></a>file_path_wide_rds <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"mosquito_wide.rds"</span>)</span>
<span id="cb54-3"><a href="#cb54-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb54-4"><a href="#cb54-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the wide-format data as an RDS file</span></span>
<span id="cb54-5"><a href="#cb54-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(wide_data, file_path_wide_rds)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
</section>
<section id="exploring-the-data" class="level1">
<h1>Exploring the Data</h1>
<p>##Study location Below is a map of the mosquito trap sites. This is an interactive map and will only work on web or other html-friendly platforms.</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb55"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb55-1"><a href="#cb55-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the map</span></span>
<span id="cb55-2"><a href="#cb55-2" aria-hidden="true" tabindex="-1"></a>map_study <span class="ot">&lt;-</span> <span class="fu">leaflet</span>(trap_location) <span class="sc">%&gt;%</span></span>
<span id="cb55-3"><a href="#cb55-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">addTiles</span>() <span class="sc">%&gt;%</span>  <span class="co"># This adds the default OpenStreetMap tiles</span></span>
<span id="cb55-4"><a href="#cb55-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">addMarkers</span>(<span class="at">lng =</span> <span class="sc">~</span>DecLong, <span class="at">lat =</span> <span class="sc">~</span>DecLat, <span class="at">popup =</span> <span class="sc">~</span>Location)  <span class="co"># Customize the popup content as needed</span></span>
<span id="cb55-5"><a href="#cb55-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb55-6"><a href="#cb55-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the map</span></span>
<span id="cb55-7"><a href="#cb55-7" aria-hidden="true" tabindex="-1"></a>map_study</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="leaflet html-widget html-fill-item" id="htmlwidget-58157d7f201a54f7f296" style="width:100%;height:464px;"></div>
<script type="application/json" data-for="htmlwidget-58157d7f201a54f7f296">{"x":{"options":{"crs":{"crsClass":"L.CRS.EPSG3857","code":null,"proj4def":null,"projectedBounds":null,"options":{}}},"calls":[{"method":"addTiles","args":["https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png",null,null,{"minZoom":0,"maxZoom":18,"tileSize":256,"subdomains":"abc","errorTileUrl":"","tms":false,"noWrap":false,"zoomOffset":0,"zoomReverse":false,"opacity":1,"zIndex":1,"detectRetina":false,"attribution":"&copy; <a href=\"https://openstreetmap.org/copyright/\">OpenStreetMap<\/a>,  <a href=\"https://opendatacommons.org/licenses/odbl/\">ODbL<\/a>"}]},{"method":"addMarkers","args":[[41.896164,41.898502,41.924639,41.906213,41.908257,41.974309,41.817419,41.812811,41.972889,41.769075,41.953882,41.771917,41.946186,41.931798,41.894792,41.765119,42.010735,41.775889,42.032111,41.72462,42.036905,41.748659,41.7501,41.748162,41.687035,41.715636,41.69198,41.822514,41.83628056,41.694168,41.859603,41.730076,41.681575,41.788811,41.669031,41.668875,42.047086,41.86055,42.004461,41.681631,42.009956,41.727154,41.677453,42.007319,41.676545,41.705864,41.67217778,41.701492,41.668303,41.908464,41.763947,41.693345,41.759094,41.728575,41.925353,41.708056,41.966909,41.685772,42.024544,41.714331,41.905447,41.72807,41.643229,41.956753,41.675197,41.9581,41.64,42.055692,41.923492,42.058381,42.05815,42.003387,41.956567,41.663933,41.66376667,41.993355,42.009242,41.933664,41.66115,41.921945,41.922767,41.98087,41.948314,41.958667,42.010128,41.980639,41.795258,41.506794,41.486575,41.828475,41.570414,41.700256,41.879161,41.866698,41.526589,41.74,41.72165,41.617931,41.63426667,41.718703,41.649858,41.631292,41.628336,41.63202778,41.761081,41.621366,41.623791,41.758331,41.735219,41.753872,41.755356,41.537222,41.66206667,41.648908,41.645669,41.665519,41.641094,41.754964,41.723778,41.65945278,41.622062,41.60659,41.649527,41.723723,41.649805,41.690266,41.62562,41.682931,41.66411667,41.64095556,41.66735,41.64633611,41.65196111,41.65749722,41.657523,41.658065,41.65372778,41.654033,41.639312,41.680218,41.68502222,41.654656,41.653862,41.606881,41.6526,41.652483,41.680659,41.648288,41.74439,41.708067,41.6443,41.678709],[-71.38026000000001,-71.37483,-71.371889,-71.367965,-71.35664800000001,-71.340733,-71.339095,-71.327223,-71.30035599999999,-71.292019,-71.28040799999999,-71.274642,-71.27351299999999,-71.26526699999999,-71.26482799999999,-71.261008,-71.234171,-71.218895,-71.21015800000001,-71.207083,-71.20413600000001,-71.198499,-71.19776400000001,-71.189562,-71.18390599999999,-71.18358000000001,-71.180587,-71.17936,-71.17890278,-71.178856,-71.17862,-71.17790599999999,-71.17277300000001,-71.171142,-71.169639,-71.16949443999999,-71.168167,-71.16801100000001,-71.167748,-71.167693,-71.16713799999999,-71.16467799999999,-71.16082400000001,-71.160608,-71.15971399999999,-71.158219,-71.1574,-71.155044,-71.15411400000001,-71.153569,-71.14966099999999,-71.146355,-71.145872,-71.13968056,-71.13963,-71.135586,-71.132773,-71.13186399999999,-71.128659,-71.12757499999999,-71.124931,-71.122292,-71.118719,-71.11579399999999,-71.109162,-71.109011,-71.10899999999999,-71.107653,-71.106669,-71.10037199999999,-71.096541,-71.096048,-71.090658,-71.08631800000001,-71.08630556,-71.07966999999999,-71.078828,-71.078689,-71.078075,-71.07708700000001,-71.076598,-71.075811,-71.070053,-71.069024,-71.063855,-71.059783,-71.057686,-71.044658,-71.037739,-71.029256,-71.018598,-71.01204199999999,-71.005942,-71.002211,-70.990861,-70.98305000000001,-70.980799,-70.98071400000001,-70.96536111,-70.962278,-70.959394,-70.958169,-70.95598099999999,-70.95593332999999,-70.95418600000001,-70.95325200000001,-70.951885,-70.95187300000001,-70.95142800000001,-70.950356,-70.948697,-70.947005,-70.94470833,-70.944294,-70.943355,-70.943077,-70.942362,-70.94234400000001,-70.941597,-70.94125556,-70.941166,-70.93980500000001,-70.93767200000001,-70.937619,-70.936814,-70.934691,-70.93275,-70.931511,-70.927725,-70.92728611,-70.926277,-70.92618889000001,-70.92576667,-70.92554167,-70.925372,-70.92219799999999,-70.92089167,-70.920461,-70.919399,-70.916591,-70.91410833,-70.914052,-70.91367200000001,-70.912325,-70.912325,-70.908992,-70.90781699999999,-70.903806,-70.89263,-70.87741699999999,-70.867653,-70.861769],null,null,null,{"interactive":true,"draggable":false,"keyboard":true,"title":"","alt":"","zIndexOffset":0,"opacity":1,"riseOnHover":false,"riseOffset":250},["Attleboro: Attleboro 02703(Bristol)","Attleboro: Attleboro 02703(Bristol)","North Attleborough: North Attleborough unknown(Bristol)","Attleboro: Attleboro 02703(Bristol)","Attleboro: Attleboro 02703(Bristol)","North Attleborough: North Attleborough 02760(Bristol)","Seekonk: Seekonk 02771(Bristol)","Seekonk: Seekonk 02771(Bristol)","North Attleborough: North Attleborough 02760(Bristol)","Swansea: Swansea 02777(Bristol)","Attleboro: Attleboro 02703(Bristol)","Swansea: Swansea 02777(Bristol)","Attleboro: Attleboro 02703(Bristol)","Attleboro: Attleboro 02703(Bristol)","Rehoboth: Rehoboth 02769(Bristol)","Swansea: Swansea 02777(Bristol)","Mansfield: Mansfield 02048(Bristol)","Swansea: Swansea 02777(Bristol)","Mansfield: Mansfield 02048(Bristol)","Swansea: Swansea 02777(Bristol)","Mansfield: Mansfield 02048(Bristol)","Swansea: Swansea 02777(Bristol)","Swansea: Swansea 02777(Bristol)","Swansea: Swansea 02777(Bristol)","Fall River: Fall River 02720(Bristol)","Somerset: Somerset 02726(Bristol)","Fall River: Fall River 02724(Bristol)","Dighton: Dighton 02715(Bristol)","Dighton: Dighton 02764(Bristol)","Fall River: Fall River 02724(Bristol)","Dighton: Dighton 02715(Bristol)","Somerset: Somerset 02725(Bristol)","Fall River: Fall River 02724(Bristol)","Swansea: Swansea 02777(Bristol)","Fall River: Fall River 02720(Bristol)","Fall River: Fall River 02721(Bristol)","Mansfield: Mansfield 02048(Bristol)","Dighton: Dighton 02715(Bristol)","Norton: Norton 02766(Bristol)","Fall River: Fall River 02720(Bristol)","Mansfield: Mansfield 02048(Bristol)","Somerset: Somerset 02725(Bristol)","Fall River: Fall River unknown(Bristol)","Norton: Norton 02766(Bristol)","Fall River: Fall River unknown(Bristol)","Fall River: Fall River unknown(Bristol)","Fall River: Fall River 02721(Bristol)","Fall River: Fall River 02720(Bristol)","Fall River: Fall River 02721(Bristol)","Taunton: Taunton 02780(Bristol)","Somerset: Somerset unknown(Bristol)","Fall River: Fall River 02720(Bristol)","Somerset: Somerset 02726(Bristol)","Fall River: Fall River 02720(Bristol)","Taunton: Taunton 02780(Bristol)","Fall River: Fall River 02720(Bristol)","Norton: Norton 02766(Bristol)","Fall River: Fall River unknown(Bristol)","Easton: Easton unknown(Bristol)","Fall River: Fall River unknown(Bristol)","Taunton: Taunton 02780(Bristol)","Fall River: Fall River 02720(Bristol)","Westport: Westport 02790(Bristol)","Taunton: Taunton 02780(Bristol)","Westport: Westport 02790(Bristol)","Taunton: Taunton 02780(Bristol)","Westport: Westport 02790(Bristol)","Easton: Easton 02356(Bristol)","Taunton: Taunton 02780(Bristol)","Easton: Easton 02356(Bristol)","Easton: Easton unknown(Bristol)","Easton: Easton 02356(Bristol)","Taunton: Taunton 02780(Bristol)","Westport: Westport 02790(Bristol)","Westport: Westport 02790(Bristol)","Easton: Easton 02375(Bristol)","Easton: Easton unknown(Bristol)","Raynham: Raynham 02767(Bristol)","Westport: Westport 02790(Bristol)","Taunton: Taunton 02780(Bristol)","Raynham: Raynham 02767(Bristol)","Raynham: Raynham unknown(Bristol)","Raynham: Raynham 02767(Bristol)","Raynham: Raynham 02767(Bristol)","Easton: Easton 02375(Bristol)","Raynham: Raynham 02767(Bristol)","Freetown: Freetown 02702(Bristol)","Westport: Westport 02790(Bristol)","Westport: Westport 02790(Bristol)","Berkley: Berkley 02779(Bristol)","Dartmouth: Dartmouth 02748(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Taunton: Taunton 02780(Bristol)","Taunton: Taunton 02780(Bristol)","Dartmouth: Dartmouth 02748(Bristol)","Freetown: Freetown unknown(Bristol)","New Bedford: New Bedford unknown(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford unknown(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Freetown: Freetown 02717(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","Dartmouth: Dartmouth 02748(Bristol)","Freetown: Freetown unknown(Bristol)","New Bedford: New Bedford 02745(Bristol)","Freetown: Freetown 02717(Bristol)","Freetown: Freetown 02717(Bristol)","Dartmouth: Dartmouth 02747(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02740(Bristol)","Freetown: Freetown unknown(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02740(Bristol)","Dartmouth: Dartmouth 02748(Bristol)","New Bedford: New Bedford unknown(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02745(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02746(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02740(Bristol)","New Bedford: New Bedford 02740(Bristol)","Acushnet: Acushnet 02743 (Bristol)","Acushnet: Acushnet 02743 (Bristol)","Fairhaven: Fairhaven 02719(Bristol)","Fairhaven: Fairhaven 02719(Bristol)","New Bedford: New Bedford unknown(Bristol)","Fairhaven: Fairhaven 02719(Bristol)","Fairhaven: Fairhaven 02719(Bristol)","Acushnet: Acushnet 02743 (Bristol)","Fairhaven: Fairhaven 02719(Bristol)","Acushnet: Acushnet 02743 (Bristol)","Acushnet: Acushnet 02743 (Bristol)","Fairhaven: Fairhaven 02719(Bristol)","Fairhaven: Fairhaven 02719(Bristol)"],null,null,null,null,{"interactive":false,"permanent":false,"direction":"auto","opacity":1,"offset":[0,0],"textsize":"10px","textOnly":false,"className":"","sticky":true},null]}],"limits":{"lat":[41.486575,42.058381],"lng":[-71.38026000000001,-70.861769]}},"evals":[],"jsHooks":[]}</script>
</div>
<div class="sourceCode cell-code" id="cb56"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb56-1"><a href="#cb56-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify the path and name of your HTML file</span></span>
<span id="cb56-2"><a href="#cb56-2" aria-hidden="true" tabindex="-1"></a>file_path_map <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"figures"</span>, <span class="st">"map_study.html"</span>)</span>
<span id="cb56-3"><a href="#cb56-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb56-4"><a href="#cb56-4" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the map</span></span>
<span id="cb56-5"><a href="#cb56-5" aria-hidden="true" tabindex="-1"></a><span class="fu">saveWidget</span>(map_study, file_path_map, <span class="at">selfcontained =</span> <span class="cn">TRUE</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<section id="eee-in-mosquitoes-and-mammals" class="level2">
<h2 class="anchored" data-anchor-id="eee-in-mosquitoes-and-mammals">EEE in mosquitoes and mammals</h2>
<section id="to-give-an-idea-of-when-eee-virus-silled-over-from-mosquitoes-and-birds-and-into-mammal-populations-lets-take-a-look-at-the-data.-lets-see-when-the-first-eee-isolations-were-found-in-mosquitoes-each-year.-lets-also-look-at-the-years-that-human-and-veterinary-cases-occured." class="level3">
<h3 class="anchored" data-anchor-id="to-give-an-idea-of-when-eee-virus-silled-over-from-mosquitoes-and-birds-and-into-mammal-populations-lets-take-a-look-at-the-data.-lets-see-when-the-first-eee-isolations-were-found-in-mosquitoes-each-year.-lets-also-look-at-the-years-that-human-and-veterinary-cases-occured.">To give an idea of when EEE virus silled over from mosquitoes and birds and into mammal populations, lets take a look at the data. Let’s see when the first EEE isolations were found in mosquitoes each year. Let’s also look at the years that human and veterinary cases occured.</h3>
</section>
<section id="from-the-table-we-see-that-human-or-othe-mammal-cases-were-recorded-in-2014-2018-2019-and-2020.-2019-had-the-highest-case-counts-for-both-humans-and-other-mammals.-because-of-this-we-will-pay-close-attention-to-2019-through-out-the-exploration." class="level3">
<h3 class="anchored" data-anchor-id="from-the-table-we-see-that-human-or-othe-mammal-cases-were-recorded-in-2014-2018-2019-and-2020.-2019-had-the-highest-case-counts-for-both-humans-and-other-mammals.-because-of-this-we-will-pay-close-attention-to-2019-through-out-the-exploration.">From the table, we see that human or othe mammal cases were recorded in 2014, 2018, 2019, and 2020. 2019 had the highest case counts for both humans and other mammals. <strong>Because of this, we will pay close attention to 2019 through out the exploration.</strong></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb57"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb57-1"><a href="#cb57-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb57-2"><a href="#cb57-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(lubridate)</span>
<span id="cb57-3"><a href="#cb57-3" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-4"><a href="#cb57-4" aria-hidden="true" tabindex="-1"></a><span class="co"># virus_data, cases_human, and cases_vet data frames </span></span>
<span id="cb57-5"><a href="#cb57-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-6"><a href="#cb57-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare virus_data: Find the first positive EEE test date by year</span></span>
<span id="cb57-7"><a href="#cb57-7" aria-hidden="true" tabindex="-1"></a>first_positive_eee_test_by_year <span class="ot">&lt;-</span> combined_virus_data <span class="sc">%&gt;%</span></span>
<span id="cb57-8"><a href="#cb57-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Virus <span class="sc">==</span> <span class="st">"EEE"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-9"><a href="#cb57-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Year =</span> <span class="fu">year</span>(<span class="fu">as.Date</span>(<span class="st">`</span><span class="at">Collection Date</span><span class="st">`</span>))) <span class="sc">%&gt;%</span></span>
<span id="cb57-10"><a href="#cb57-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Year) <span class="sc">%&gt;%</span></span>
<span id="cb57-11"><a href="#cb57-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">First_Positive_Test_Date =</span> <span class="fu">min</span>(<span class="st">`</span><span class="at">Collection Date</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-12"><a href="#cb57-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-13"><a href="#cb57-13" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Year)</span>
<span id="cb57-14"><a href="#cb57-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-15"><a href="#cb57-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare cases_human: Summarize human EEE cases by year</span></span>
<span id="cb57-16"><a href="#cb57-16" aria-hidden="true" tabindex="-1"></a>human_cases_by_year <span class="ot">&lt;-</span> cases_human <span class="sc">%&gt;%</span></span>
<span id="cb57-17"><a href="#cb57-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(<span class="st">`</span><span class="at">Virus Result</span><span class="st">`</span> <span class="sc">==</span> <span class="st">"EEE"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-18"><a href="#cb57-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Year =</span> <span class="fu">year</span>(<span class="st">`</span><span class="at">Onset Date</span><span class="st">`</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-19"><a href="#cb57-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Year) <span class="sc">%&gt;%</span></span>
<span id="cb57-20"><a href="#cb57-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Human_Cases =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb57-21"><a href="#cb57-21" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-22"><a href="#cb57-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(Year)</span>
<span id="cb57-23"><a href="#cb57-23" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-24"><a href="#cb57-24" aria-hidden="true" tabindex="-1"></a><span class="co"># Prepare cases_vet: Summarize animal EEE cases by year</span></span>
<span id="cb57-25"><a href="#cb57-25" aria-hidden="true" tabindex="-1"></a>animal_cases_by_year <span class="ot">&lt;-</span> cases_vet <span class="sc">%&gt;%</span></span>
<span id="cb57-26"><a href="#cb57-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(virus <span class="sc">==</span> <span class="st">"eee"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-27"><a href="#cb57-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(year) <span class="sc">%&gt;%</span></span>
<span id="cb57-28"><a href="#cb57-28" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Animal_Cases =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span></span>
<span id="cb57-29"><a href="#cb57-29" aria-hidden="true" tabindex="-1"></a>  <span class="fu">ungroup</span>() <span class="sc">%&gt;%</span></span>
<span id="cb57-30"><a href="#cb57-30" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(year)</span>
<span id="cb57-31"><a href="#cb57-31" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-32"><a href="#cb57-32" aria-hidden="true" tabindex="-1"></a><span class="co"># Ensure 'Year' column is numeric across all data frames for compatibility</span></span>
<span id="cb57-33"><a href="#cb57-33" aria-hidden="true" tabindex="-1"></a><span class="co"># Note: This step might already be covered by the mutations above but included here for clarity</span></span>
<span id="cb57-34"><a href="#cb57-34" aria-hidden="true" tabindex="-1"></a>first_positive_eee_test_by_year<span class="sc">$</span>Year <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(first_positive_eee_test_by_year<span class="sc">$</span>Year)</span>
<span id="cb57-35"><a href="#cb57-35" aria-hidden="true" tabindex="-1"></a>animal_cases_by_year<span class="sc">$</span>year <span class="ot">&lt;-</span> <span class="fu">as.numeric</span>(animal_cases_by_year<span class="sc">$</span>year) <span class="co"># Make sure year is numeric and named consistently</span></span>
<span id="cb57-36"><a href="#cb57-36" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-37"><a href="#cb57-37" aria-hidden="true" tabindex="-1"></a><span class="co"># Merge the summaries into one combined data frame</span></span>
<span id="cb57-38"><a href="#cb57-38" aria-hidden="true" tabindex="-1"></a>combined_eee_data <span class="ot">&lt;-</span> first_positive_eee_test_by_year <span class="sc">%&gt;%</span></span>
<span id="cb57-39"><a href="#cb57-39" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(human_cases_by_year, <span class="at">by =</span> <span class="st">"Year"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-40"><a href="#cb57-40" aria-hidden="true" tabindex="-1"></a>  <span class="fu">left_join</span>(animal_cases_by_year, <span class="at">by =</span> <span class="fu">c</span>(<span class="st">"Year"</span> <span class="ot">=</span> <span class="st">"year"</span>)) <span class="co"># Ensure correct column names are used for joining</span></span>
<span id="cb57-41"><a href="#cb57-41" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-42"><a href="#cb57-42" aria-hidden="true" tabindex="-1"></a><span class="co"># Replace NA values with 0 for Human_Cases and Animal_Cases</span></span>
<span id="cb57-43"><a href="#cb57-43" aria-hidden="true" tabindex="-1"></a>combined_eee_data<span class="sc">$</span>Human_Cases[<span class="fu">is.na</span>(combined_eee_data<span class="sc">$</span>Human_Cases)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb57-44"><a href="#cb57-44" aria-hidden="true" tabindex="-1"></a>combined_eee_data<span class="sc">$</span>Animal_Cases[<span class="fu">is.na</span>(combined_eee_data<span class="sc">$</span>Animal_Cases)] <span class="ot">&lt;-</span> <span class="dv">0</span></span>
<span id="cb57-45"><a href="#cb57-45" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-46"><a href="#cb57-46" aria-hidden="true" tabindex="-1"></a><span class="co"># Use kable() to create a basic table</span></span>
<span id="cb57-47"><a href="#cb57-47" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(knitr)</span>
<span id="cb57-48"><a href="#cb57-48" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(kableExtra)</span>
<span id="cb57-49"><a href="#cb57-49" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-50"><a href="#cb57-50" aria-hidden="true" tabindex="-1"></a><span class="co"># Creating the table with centered headings and data</span></span>
<span id="cb57-51"><a href="#cb57-51" aria-hidden="true" tabindex="-1"></a>kable_table_isolation_case <span class="ot">&lt;-</span> <span class="fu">kable</span>(combined_eee_data, <span class="st">"html"</span>, </span>
<span id="cb57-52"><a href="#cb57-52" aria-hidden="true" tabindex="-1"></a>                     <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"Year"</span>, <span class="st">"First Positive EEE Test Date"</span>, <span class="st">"Human EEE Cases"</span>, <span class="st">"Animal EEE Cases"</span>),</span>
<span id="cb57-53"><a href="#cb57-53" aria-hidden="true" tabindex="-1"></a>                     <span class="at">caption =</span> <span class="st">"Annual Summary of EEE Virus Activity"</span>,</span>
<span id="cb57-54"><a href="#cb57-54" aria-hidden="true" tabindex="-1"></a>                     <span class="at">align =</span> <span class="fu">c</span>(<span class="st">'c'</span>,<span class="st">'c'</span>,<span class="st">'c'</span>,<span class="st">'c'</span>)) <span class="sc">%&gt;%</span> <span class="co"># This aligns all columns' data to center</span></span>
<span id="cb57-55"><a href="#cb57-55" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>, <span class="st">"condensed"</span>, <span class="st">"responsive"</span>),</span>
<span id="cb57-56"><a href="#cb57-56" aria-hidden="true" tabindex="-1"></a>                <span class="at">full_width =</span> F, <span class="at">position =</span> <span class="st">"center"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-57"><a href="#cb57-57" aria-hidden="true" tabindex="-1"></a>  <span class="fu">add_header_above</span>(<span class="fu">c</span>(<span class="st">" "</span> <span class="ot">=</span> <span class="dv">1</span>, <span class="st">"EEE Virus Detection and Cases"</span> <span class="ot">=</span> <span class="dv">3</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb57-58"><a href="#cb57-58" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-59"><a href="#cb57-59" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span>, <span class="at">width =</span> <span class="st">"8em"</span>) <span class="sc">%&gt;%</span> <span class="co"># Adjusted width without altering alignment</span></span>
<span id="cb57-60"><a href="#cb57-60" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">3</span>, <span class="at">width =</span> <span class="st">"5em"</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"red"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb57-61"><a href="#cb57-61" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">4</span>, <span class="at">width =</span> <span class="st">"5em"</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"blue"</span>)</span>
<span id="cb57-62"><a href="#cb57-62" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb57-63"><a href="#cb57-63" aria-hidden="true" tabindex="-1"></a><span class="co"># Displaying the table</span></span>
<span id="cb57-64"><a href="#cb57-64" aria-hidden="true" tabindex="-1"></a>kable_table_isolation_case</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-striped table-hover table-condensed table-sm small" data-quarto-postprocess="true">
<caption>Annual Summary of EEE Virus Activity</caption>
<colgroup>
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
<col style="width: 25%">
</colgroup>
<thead>
<tr class="header">
<th data-quarto-table-cell-role="th" style="text-align: center; empty-cells: hide; border-bottom: hidden;"></th>
<th colspan="3" data-quarto-table-cell-role="th" style="text-align: center; border-bottom: hidden; padding-bottom: 0; padding-left: 3px; padding-right: 3px;"><div style="border-bottom: 1px solid #ddd; padding-bottom: 5px; ">
EEE Virus Detection and Cases
</div></th>
</tr>
<tr class="odd">
<th style="text-align: center;" data-quarto-table-cell-role="th">Year</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">First Positive EEE Test Date</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">Human EEE Cases</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">Animal EEE Cases</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">2014</td>
<td style="text-align: center; width: 8em;">2014-07-15</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">0</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">2</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">2015</td>
<td style="text-align: center; width: 8em;">2015-09-25</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">0</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">0</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">2016</td>
<td style="text-align: center; width: 8em;">2016-07-12</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">0</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">0</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">2017</td>
<td style="text-align: center; width: 8em;">2017-08-14</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">0</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">0</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">2018</td>
<td style="text-align: center; width: 8em;">2018-09-17</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">0</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">2</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">2019</td>
<td style="text-align: center; width: 8em;">2019-07-15</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">12</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">9</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">2020</td>
<td style="text-align: center; width: 8em;">2020-07-01</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: red !important;">5</td>
<td style="text-align: center; width: 5em; font-weight: bold; color: blue !important;">0</td>
</tr>
</tbody>
</table>
</div>


</div>
<div class="sourceCode cell-code" id="cb58"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb58-1"><a href="#cb58-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path where you want to save the HTML file</span></span>
<span id="cb58-2"><a href="#cb58-2" aria-hidden="true" tabindex="-1"></a><span class="co">#output_path_quarto &lt;- here("results", "tables", "eee_isolation_cases")</span></span>
<span id="cb58-3"><a href="#cb58-3" aria-hidden="true" tabindex="-1"></a>output_path_png <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"tables"</span>, <span class="st">"eee_isolation_cases.png"</span>)</span>
<span id="cb58-4"><a href="#cb58-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-5"><a href="#cb58-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the kableExtra table as an HTML file</span></span>
<span id="cb58-6"><a href="#cb58-6" aria-hidden="true" tabindex="-1"></a><span class="co">#save_kable(kable_table_isolation_case, file = output_path_quarto)</span></span>
<span id="cb58-7"><a href="#cb58-7" aria-hidden="true" tabindex="-1"></a><span class="fu">save_kable</span>(kable_table_isolation_case, <span class="at">file =</span> output_path_png)</span>
<span id="cb58-8"><a href="#cb58-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-9"><a href="#cb58-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the paths for the RDS files</span></span>
<span id="cb58-10"><a href="#cb58-10" aria-hidden="true" tabindex="-1"></a>first_positive_eee_test_by_year_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"first_positive_eee_test_by_year.rds"</span>)</span>
<span id="cb58-11"><a href="#cb58-11" aria-hidden="true" tabindex="-1"></a>human_cases_by_year_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"human_cases_by_year.rds"</span>)</span>
<span id="cb58-12"><a href="#cb58-12" aria-hidden="true" tabindex="-1"></a>animal_cases_by_year_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"animal_cases_by_year.rds"</span>)</span>
<span id="cb58-13"><a href="#cb58-13" aria-hidden="true" tabindex="-1"></a>combined_eee_data_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"combined_eee_data.rds"</span>)</span>
<span id="cb58-14"><a href="#cb58-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb58-15"><a href="#cb58-15" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the data frames as RDS files</span></span>
<span id="cb58-16"><a href="#cb58-16" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(first_positive_eee_test_by_year, first_positive_eee_test_by_year_path)</span>
<span id="cb58-17"><a href="#cb58-17" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(human_cases_by_year, human_cases_by_year_path)</span>
<span id="cb58-18"><a href="#cb58-18" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(animal_cases_by_year, animal_cases_by_year_path)</span>
<span id="cb58-19"><a href="#cb58-19" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(combined_eee_data, combined_eee_data_path)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
</section>
<section id="which-county-has-the-most-eee-positive-mosquitoes-of-those-tested" class="level2">
<h2 class="anchored" data-anchor-id="which-county-has-the-most-eee-positive-mosquitoes-of-those-tested">Which county has the most EEE positive mosquitoes of those tested?</h2>
<div class="cell">
<div class="sourceCode cell-code" id="cb59"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb59-1"><a href="#cb59-1" aria-hidden="true" tabindex="-1"></a>eee_positive_by_county <span class="ot">&lt;-</span> combined_virus_data <span class="sc">%&gt;%</span></span>
<span id="cb59-2"><a href="#cb59-2" aria-hidden="true" tabindex="-1"></a>  <span class="fu">filter</span>(Virus <span class="sc">==</span> <span class="st">"EEE"</span>) <span class="sc">%&gt;%</span>  <span class="co"># Filter for rows where Virus is EEE</span></span>
<span id="cb59-3"><a href="#cb59-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(County) <span class="sc">%&gt;%</span></span>
<span id="cb59-4"><a href="#cb59-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">EEE_Positive_Tests =</span> <span class="fu">n</span>()) <span class="sc">%&gt;%</span>  <span class="co"># Count the number of EEE positive tests in each group</span></span>
<span id="cb59-5"><a href="#cb59-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">arrange</span>(<span class="fu">desc</span>(EEE_Positive_Tests))  <span class="co"># Order the results by EEE_Positive_Tests in descending order</span></span>
<span id="cb59-6"><a href="#cb59-6" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-7"><a href="#cb59-7" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the table with kable and style it with kableExtra</span></span>
<span id="cb59-8"><a href="#cb59-8" aria-hidden="true" tabindex="-1"></a>eee_positive_table <span class="ot">&lt;-</span> <span class="fu">kable</span>(eee_positive_by_county, <span class="st">"html"</span>, </span>
<span id="cb59-9"><a href="#cb59-9" aria-hidden="true" tabindex="-1"></a>                            <span class="at">col.names =</span> <span class="fu">c</span>(<span class="st">"County"</span>, <span class="st">"EEE Positive Tests"</span>),</span>
<span id="cb59-10"><a href="#cb59-10" aria-hidden="true" tabindex="-1"></a>                            <span class="at">caption =</span> <span class="st">"EEE Positive Tests by County"</span>,</span>
<span id="cb59-11"><a href="#cb59-11" aria-hidden="true" tabindex="-1"></a>                            <span class="at">align =</span> <span class="fu">c</span>(<span class="st">'c'</span>,<span class="st">'c'</span>)) <span class="sc">%&gt;%</span></span>
<span id="cb59-12"><a href="#cb59-12" aria-hidden="true" tabindex="-1"></a>  <span class="fu">kable_styling</span>(<span class="at">bootstrap_options =</span> <span class="fu">c</span>(<span class="st">"striped"</span>, <span class="st">"hover"</span>, <span class="st">"condensed"</span>, <span class="st">"responsive"</span>),</span>
<span id="cb59-13"><a href="#cb59-13" aria-hidden="true" tabindex="-1"></a>                <span class="at">full_width =</span> <span class="cn">FALSE</span>, <span class="at">position =</span> <span class="st">"center"</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-14"><a href="#cb59-14" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">1</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>) <span class="sc">%&gt;%</span></span>
<span id="cb59-15"><a href="#cb59-15" aria-hidden="true" tabindex="-1"></a>  <span class="fu">column_spec</span>(<span class="dv">2</span>, <span class="at">bold =</span> <span class="cn">TRUE</span>, <span class="at">color =</span> <span class="st">"red"</span>)</span>
<span id="cb59-16"><a href="#cb59-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb59-17"><a href="#cb59-17" aria-hidden="true" tabindex="-1"></a><span class="co"># Display the table</span></span>
<span id="cb59-18"><a href="#cb59-18" aria-hidden="true" tabindex="-1"></a>eee_positive_table</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div class="table-responsive">
<table class="table table-striped table-hover table-condensed table-sm small" data-quarto-postprocess="true">
<caption>EEE Positive Tests by County</caption>
<thead>
<tr class="header">
<th style="text-align: center;" data-quarto-table-cell-role="th">County</th>
<th style="text-align: center;" data-quarto-table-cell-role="th">EEE Positive Tests</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Plymouth</td>
<td style="text-align: center; font-weight: bold; color: red !important;">243</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">Bristol</td>
<td style="text-align: center; font-weight: bold; color: red !important;">177</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Worcester</td>
<td style="text-align: center; font-weight: bold; color: red !important;">48</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">Norfolk</td>
<td style="text-align: center; font-weight: bold; color: red !important;">17</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Barnstable</td>
<td style="text-align: center; font-weight: bold; color: red !important;">16</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">Essex</td>
<td style="text-align: center; font-weight: bold; color: red !important;">13</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Middlesex</td>
<td style="text-align: center; font-weight: bold; color: red !important;">12</td>
</tr>
<tr class="even">
<td style="text-align: center; font-weight: bold;">Hampden</td>
<td style="text-align: center; font-weight: bold; color: red !important;">7</td>
</tr>
<tr class="odd">
<td style="text-align: center; font-weight: bold;">Franklin</td>
<td style="text-align: center; font-weight: bold; color: red !important;">2</td>
</tr>
</tbody>
</table>
</div>


</div>
<div class="sourceCode cell-code" id="cb60"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb60-1"><a href="#cb60-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path where you want to save the HTML file</span></span>
<span id="cb60-2"><a href="#cb60-2" aria-hidden="true" tabindex="-1"></a><span class="co">#output_path_qmd &lt;- here("results", "tables", "eee_isolation_county.html")</span></span>
<span id="cb60-3"><a href="#cb60-3" aria-hidden="true" tabindex="-1"></a>output_path_county <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"tables"</span>, <span class="st">"eee_isolation_county.html"</span>)</span>
<span id="cb60-4"><a href="#cb60-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb60-5"><a href="#cb60-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the kableExtra table as an HTML file</span></span>
<span id="cb60-6"><a href="#cb60-6" aria-hidden="true" tabindex="-1"></a><span class="co">#save_kable(eee_positive_table, file = output_path_qmd)</span></span>
<span id="cb60-7"><a href="#cb60-7" aria-hidden="true" tabindex="-1"></a><span class="fu">save_kable</span>(eee_positive_table, <span class="at">file =</span> output_path_county)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
</section>
<section id="culiseta-melanura-mel-is-the-main-enzootic-vector-of-eee.-some-entomologist-believe-that-high-levels-of-mel-traps-indicate-higher-risk-for-eee-transmission.-using-the-new-mosquito-collection-table-created-from-the-pivot-lets-calculate-the-proportion-of-mel-captured-compared-to-all-other-mosquito-vector-species-and-create-a-plot-to-visualize-it." class="level2">
<h2 class="anchored" data-anchor-id="culiseta-melanura-mel-is-the-main-enzootic-vector-of-eee.-some-entomologist-believe-that-high-levels-of-mel-traps-indicate-higher-risk-for-eee-transmission.-using-the-new-mosquito-collection-table-created-from-the-pivot-lets-calculate-the-proportion-of-mel-captured-compared-to-all-other-mosquito-vector-species-and-create-a-plot-to-visualize-it."><em>Culiseta melanura</em> (MEL) is the main enzootic vector of EEE. Some entomologist believe that high levels of MEL traps indicate higher risk for EEE transmission. Using the new mosquito collection table created from the pivot, let’s calculate the proportion of MEL captured compared to all other mosquito vector species and create a plot to visualize it.</h2>
<section id="section" class="level3">
<h3 class="anchored" data-anchor-id="section"></h3>
<div class="cell">
<div class="sourceCode cell-code" id="cb61"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb61-1"><a href="#cb61-1" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(dplyr)</span>
<span id="cb61-2"><a href="#cb61-2" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(ggplot2)</span>
<span id="cb61-3"><a href="#cb61-3" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(readr) <span class="co"># For read_csv</span></span>
<span id="cb61-4"><a href="#cb61-4" aria-hidden="true" tabindex="-1"></a><span class="fu">library</span>(here)</span>
<span id="cb61-5"><a href="#cb61-5" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-6"><a href="#cb61-6" aria-hidden="true" tabindex="-1"></a><span class="co"># Define path to the data file</span></span>
<span id="cb61-7"><a href="#cb61-7" aria-hidden="true" tabindex="-1"></a>trap_wide_path <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"mosquito_wide.csv"</span>)</span>
<span id="cb61-8"><a href="#cb61-8" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb61-9"><a href="#cb61-9" aria-hidden="true" tabindex="-1"></a><span class="co"># Read the data</span></span>
<span id="cb61-10"><a href="#cb61-10" aria-hidden="true" tabindex="-1"></a>trap_proportion <span class="ot">&lt;-</span> <span class="fu">read_csv</span>(trap_wide_path) <span class="sc">%&gt;%</span></span>
<span id="cb61-11"><a href="#cb61-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">mutate</span>(<span class="at">Year =</span> <span class="fu">year</span>(<span class="fu">as.Date</span>(<span class="st">`</span><span class="at">collection date</span><span class="st">`</span>, <span class="at">format =</span> <span class="st">"%Y-%m-%d"</span>)))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Rows: 5109 Columns: 55
── Column specification ────────────────────────────────────────────────────────
Delimiter: ","
chr   (3): town, trap type, mcd
dbl  (51): ABS, CAN, GRO, MEL, PUN, EXC, QUA, AUR, STI, TRI, VEX, RES, JAP, ...
date  (1): collection date

ℹ Use `spec()` to retrieve the full column specification for this data.
ℹ Specify the column types or set `show_col_types = FALSE` to quiet this message.</code></pre>
</div>
<div class="sourceCode cell-code" id="cb63"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb63-1"><a href="#cb63-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the proportion of MEL out of the total for all species</span></span>
<span id="cb63-2"><a href="#cb63-2" aria-hidden="true" tabindex="-1"></a>proportion_mel_by_year <span class="ot">&lt;-</span> trap_proportion <span class="sc">%&gt;%</span></span>
<span id="cb63-3"><a href="#cb63-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Year) <span class="sc">%&gt;%</span></span>
<span id="cb63-4"><a href="#cb63-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Total_MEL =</span> <span class="fu">sum</span>(MEL, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-5"><a href="#cb63-5" aria-hidden="true" tabindex="-1"></a>            <span class="at">Total_All =</span> <span class="fu">sum</span>(MEL <span class="sc">+</span> PER <span class="sc">+</span> PIP <span class="sc">+</span> RES <span class="sc">+</span> SAL <span class="sc">+</span> TRI <span class="sc">+</span> SAP, <span class="at">na.rm =</span> <span class="cn">TRUE</span>),</span>
<span id="cb63-6"><a href="#cb63-6" aria-hidden="true" tabindex="-1"></a>            <span class="at">MEL_Prop =</span> Total_MEL <span class="sc">/</span> Total_All)</span>
<span id="cb63-7"><a href="#cb63-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-8"><a href="#cb63-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Calculate the average number of MEL by year</span></span>
<span id="cb63-9"><a href="#cb63-9" aria-hidden="true" tabindex="-1"></a>average_mel_by_year <span class="ot">&lt;-</span> trap_proportion <span class="sc">%&gt;%</span></span>
<span id="cb63-10"><a href="#cb63-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">group_by</span>(Year) <span class="sc">%&gt;%</span></span>
<span id="cb63-11"><a href="#cb63-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">summarise</span>(<span class="at">Average_MEL =</span> <span class="fu">mean</span>(MEL, <span class="at">na.rm =</span> <span class="cn">TRUE</span>))</span>
<span id="cb63-12"><a href="#cb63-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-13"><a href="#cb63-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the processed data as RDS and CSV</span></span>
<span id="cb63-14"><a href="#cb63-14" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(proportion_mel_by_year, <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"yearly_mel_proportion.rds"</span>))</span>
<span id="cb63-15"><a href="#cb63-15" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(proportion_mel_by_year, <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"yearly_mel_proportion.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb63-16"><a href="#cb63-16" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-17"><a href="#cb63-17" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(average_mel_by_year, <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"yearly_mel_average.rds"</span>))</span>
<span id="cb63-18"><a href="#cb63-18" aria-hidden="true" tabindex="-1"></a><span class="fu">write.csv</span>(average_mel_by_year, <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"yearly_mel_average.csv"</span>), <span class="at">row.names =</span> <span class="cn">FALSE</span>)</span>
<span id="cb63-19"><a href="#cb63-19" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb63-20"><a href="#cb63-20" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot for the proportion of MEL by year</span></span>
<span id="cb63-21"><a href="#cb63-21" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(proportion_mel_by_year, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Year), <span class="at">y =</span> MEL_Prop)) <span class="sc">+</span></span>
<span id="cb63-22"><a href="#cb63-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb63-23"><a href="#cb63-23" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Proportion of Culiseta melanura (MEL) by Year"</span>,</span>
<span id="cb63-24"><a href="#cb63-24" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Year"</span>,</span>
<span id="cb63-25"><a href="#cb63-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Proportion of MEL"</span>) <span class="sc">+</span></span>
<span id="cb63-26"><a href="#cb63-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb63-27"><a href="#cb63-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="MADAproject_pt2_ruiz_files/figure-html/unnamed-chunk-16-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb64"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb64-1"><a href="#cb64-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"figures"</span>, <span class="st">"mel_proportion_by_year.png"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span>
<span id="cb64-2"><a href="#cb64-2" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb64-3"><a href="#cb64-3" aria-hidden="true" tabindex="-1"></a><span class="co"># Plot for the average count of MEL by year</span></span>
<span id="cb64-4"><a href="#cb64-4" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(average_mel_by_year, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Year), <span class="at">y =</span> Average_MEL, <span class="at">fill =</span> <span class="st">"Year"</span>)) <span class="sc">+</span></span>
<span id="cb64-5"><a href="#cb64-5" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>() <span class="sc">+</span></span>
<span id="cb64-6"><a href="#cb64-6" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Average Count of Culiseta melanura (MEL) by Year"</span>,</span>
<span id="cb64-7"><a href="#cb64-7" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Year"</span>,</span>
<span id="cb64-8"><a href="#cb64-8" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Average Count of MEL"</span>) <span class="sc">+</span></span>
<span id="cb64-9"><a href="#cb64-9" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_fill_viridis_d</span>() <span class="sc">+</span></span>
<span id="cb64-10"><a href="#cb64-10" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb64-11"><a href="#cb64-11" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="MADAproject_pt2_ruiz_files/figure-html/unnamed-chunk-16-2.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb65"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb65-1"><a href="#cb65-1" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"figures"</span>, <span class="st">"average_mel_by_year.png"</span>), <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
</div>
<p>ggsave(file_path_avg_mel, plot = mel_avg_plot, width = 8, height = 6, dpi = 300)</p>
<pre><code>
## Mosquito abundance is affected by weather. Mosquitoes depend on standing water to complete their larval development. Let's take a look at the annual precipitation using the wx data frame.

::: {.cell}

```{.r .cell-code}
# Convert DATE from character to Date type using as.Date and specifying the format
wx$DATE &lt;- as.Date(wx$DATE, format="%m/%d/%y")

# The as.Date function treats years in the format "yy" as follows:
# Years 00-68 are treated as 2000-2068, and years 69-99 are treated as 1969-1999.
# Adjust this according to your data's actual year range if needed.

# Extract the year from each DATE
wx$Year &lt;- format(wx$DATE, "%Y") %&gt;% as.numeric()

# Now that we have the years as numeric values, let's filter for 2013 to 2020
filtered_wx &lt;- wx %&gt;%
  filter(Year &gt;= 2013 &amp; Year &lt;= 2020)

# Save the filtered weather data as an RDS file
filtered_wx_path_rds &lt;- here("data", "processed-data", "rds", "filtered_wx_2013_2020.rds")
saveRDS(filtered_wx, filtered_wx_path_rds)

# Sum PRCP by Year for the filtered data
summed_prcp_by_year_filtered &lt;- filtered_wx %&gt;%
  group_by(Year) %&gt;%
  summarise(Total_PRCP = sum(PRCP, na.rm = TRUE))

# Save the summarized precipitation data as an RDS file
summed_prcp_by_year_path_rds &lt;- here("data", "processed-data", "rds", "summed_prcp_by_year_filtered_2013_2020.rds")
saveRDS(summed_prcp_by_year_filtered, summed_prcp_by_year_path_rds)

# View the result
print(summed_prcp_by_year_filtered)</code></pre>
<div class="cell-output cell-output-stdout">
<pre><code># A tibble: 8 × 2
   Year Total_PRCP
  &lt;dbl&gt;      &lt;dbl&gt;
1  2013       44.1
2  2014       49.1
3  2015       40.8
4  2016       34.6
5  2017       43.1
6  2018       62.0
7  2019       50.0
8  2020       41.8</code></pre>
</div>
<div class="sourceCode cell-code" id="cb68"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb68-1"><a href="#cb68-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Create the ggplot object and assign it to a variable</span></span>
<span id="cb68-2"><a href="#cb68-2" aria-hidden="true" tabindex="-1"></a>precip_plot <span class="ot">&lt;-</span> <span class="fu">ggplot</span>(summed_prcp_by_year_filtered, <span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">as.factor</span>(Year), <span class="at">y =</span> Total_PRCP)) <span class="sc">+</span></span>
<span id="cb68-3"><a href="#cb68-3" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_bar</span>(<span class="at">stat =</span> <span class="st">"identity"</span>, <span class="at">fill =</span> <span class="st">"steelblue"</span>) <span class="sc">+</span></span>
<span id="cb68-4"><a href="#cb68-4" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Total Precipitation by Year (2013-2020)"</span>,</span>
<span id="cb68-5"><a href="#cb68-5" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Year"</span>,</span>
<span id="cb68-6"><a href="#cb68-6" aria-hidden="true" tabindex="-1"></a>       <span class="at">y =</span> <span class="st">"Total Precipitation (inches)"</span>) <span class="sc">+</span></span>
<span id="cb68-7"><a href="#cb68-7" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb68-8"><a href="#cb68-8" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>))</span>
<span id="cb68-9"><a href="#cb68-9" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-10"><a href="#cb68-10" aria-hidden="true" tabindex="-1"></a><span class="co"># Specify the file path where you want to save the plot</span></span>
<span id="cb68-11"><a href="#cb68-11" aria-hidden="true" tabindex="-1"></a>file_path_precip <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"figures"</span>, <span class="st">"annual_precip.png"</span>)</span>
<span id="cb68-12"><a href="#cb68-12" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb68-13"><a href="#cb68-13" aria-hidden="true" tabindex="-1"></a><span class="co"># Use ggsave to save the plot, make sure to use the plot object</span></span>
<span id="cb68-14"><a href="#cb68-14" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(file_path_precip, <span class="at">plot =</span> precip_plot, <span class="at">width =</span> <span class="dv">8</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<p>::: ### Now I want to plot the ave mel by year with the total precip by year</p>
<div class="cell">
<div class="sourceCode cell-code" id="cb69"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb69-1"><a href="#cb69-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Load the datasets</span></span>
<span id="cb69-2"><a href="#cb69-2" aria-hidden="true" tabindex="-1"></a>yearly_mel_ave <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"yearly_mel_ave.rds"</span>))</span>
<span id="cb69-3"><a href="#cb69-3" aria-hidden="true" tabindex="-1"></a>summed_prcp_by_year_filtered <span class="ot">&lt;-</span> <span class="fu">readRDS</span>(<span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"summed_prcp_by_year_filtered_2013_2020.rds"</span>))</span>
<span id="cb69-4"><a href="#cb69-4" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-5"><a href="#cb69-5" aria-hidden="true" tabindex="-1"></a><span class="co"># Combine the datasets</span></span>
<span id="cb69-6"><a href="#cb69-6" aria-hidden="true" tabindex="-1"></a>combined_data <span class="ot">&lt;-</span> <span class="fu">left_join</span>(yearly_mel_ave, summed_prcp_by_year_filtered, <span class="at">by =</span> <span class="st">"Year"</span>)</span>
<span id="cb69-7"><a href="#cb69-7" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-8"><a href="#cb69-8" aria-hidden="true" tabindex="-1"></a><span class="co"># Define the path for the RDS file with a descriptive name</span></span>
<span id="cb69-9"><a href="#cb69-9" aria-hidden="true" tabindex="-1"></a>combined_data_path_rds <span class="ot">&lt;-</span> <span class="fu">here</span>(<span class="st">"data"</span>, <span class="st">"processed-data"</span>, <span class="st">"rds"</span>, <span class="st">"combined_mel_avg_and_total_precip_2013_2020.rds"</span>)</span>
<span id="cb69-10"><a href="#cb69-10" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-11"><a href="#cb69-11" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the combined data as an RDS file</span></span>
<span id="cb69-12"><a href="#cb69-12" aria-hidden="true" tabindex="-1"></a><span class="fu">saveRDS</span>(combined_data, combined_data_path_rds)</span>
<span id="cb69-13"><a href="#cb69-13" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-14"><a href="#cb69-14" aria-hidden="true" tabindex="-1"></a></span>
<span id="cb69-15"><a href="#cb69-15" aria-hidden="true" tabindex="-1"></a><span class="fu">ggplot</span>(<span class="at">data =</span> combined_data) <span class="sc">+</span></span>
<span id="cb69-16"><a href="#cb69-16" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_col</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Year), <span class="at">y =</span> Average_MEL), <span class="at">fill =</span> <span class="st">"blue"</span>) <span class="sc">+</span></span>
<span id="cb69-17"><a href="#cb69-17" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_line</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Year), <span class="at">y =</span> Total_PRCP, <span class="at">group =</span> <span class="dv">1</span>), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">linetype =</span> <span class="st">"dashed"</span>) <span class="sc">+</span></span>
<span id="cb69-18"><a href="#cb69-18" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_point</span>(<span class="fu">aes</span>(<span class="at">x =</span> <span class="fu">factor</span>(Year), <span class="at">y =</span> Total_PRCP), <span class="at">color =</span> <span class="st">"red"</span>, <span class="at">size =</span> <span class="dv">3</span>) <span class="sc">+</span></span>
<span id="cb69-19"><a href="#cb69-19" aria-hidden="true" tabindex="-1"></a>  <span class="fu">geom_hline</span>(<span class="at">yintercept =</span> <span class="fl">45.6</span>, <span class="at">linetype =</span> <span class="st">"dotted"</span>, <span class="at">color =</span> <span class="st">"darkgreen"</span>, <span class="at">size =</span> <span class="dv">1</span>) <span class="sc">+</span></span>
<span id="cb69-20"><a href="#cb69-20" aria-hidden="true" tabindex="-1"></a>  <span class="fu">annotate</span>(<span class="st">"text"</span>, <span class="at">x =</span> <span class="cn">Inf</span>, <span class="at">y =</span> <span class="fl">45.6</span>, <span class="at">label =</span> <span class="st">"Avg Precip: 45.6 inches"</span>, </span>
<span id="cb69-21"><a href="#cb69-21" aria-hidden="true" tabindex="-1"></a>           <span class="at">hjust =</span> <span class="fl">3.8</span>, <span class="at">vjust =</span> <span class="sc">-</span><span class="fl">1.1</span>, <span class="at">color =</span> <span class="st">"darkgreen"</span>, <span class="at">size =</span> <span class="fl">3.5</span>, <span class="at">face =</span> <span class="st">"bold"</span>, <span class="at">angle =</span> <span class="dv">0</span>) <span class="sc">+</span></span>
<span id="cb69-22"><a href="#cb69-22" aria-hidden="true" tabindex="-1"></a>  <span class="fu">scale_y_continuous</span>(<span class="at">name =</span> <span class="st">"Average Count of MEL"</span>, </span>
<span id="cb69-23"><a href="#cb69-23" aria-hidden="true" tabindex="-1"></a>                     <span class="at">sec.axis =</span> <span class="fu">sec_axis</span>(<span class="sc">~</span> ., <span class="at">name =</span> <span class="st">"Total Precipitation (inches)"</span>)) <span class="sc">+</span></span>
<span id="cb69-24"><a href="#cb69-24" aria-hidden="true" tabindex="-1"></a>  <span class="fu">labs</span>(<span class="at">title =</span> <span class="st">"Average Count of MEL Mosquitoes and Total Precipitation by Year"</span>,</span>
<span id="cb69-25"><a href="#cb69-25" aria-hidden="true" tabindex="-1"></a>       <span class="at">x =</span> <span class="st">"Year"</span>) <span class="sc">+</span></span>
<span id="cb69-26"><a href="#cb69-26" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme_minimal</span>() <span class="sc">+</span></span>
<span id="cb69-27"><a href="#cb69-27" aria-hidden="true" tabindex="-1"></a>  <span class="fu">theme</span>(<span class="at">axis.text.x =</span> <span class="fu">element_text</span>(<span class="at">angle =</span> <span class="dv">45</span>, <span class="at">hjust =</span> <span class="dv">1</span>, <span class="at">size =</span> <span class="dv">12</span>, <span class="at">face =</span> <span class="st">"bold"</span>),</span>
<span id="cb69-28"><a href="#cb69-28" aria-hidden="true" tabindex="-1"></a>        <span class="at">axis.title.x =</span> <span class="fu">element_text</span>(<span class="at">face =</span> <span class="st">"bold"</span>))</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Using `size` aesthetic for lines was deprecated in ggplot2 3.4.0.
ℹ Please use `linewidth` instead.</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning in annotate("text", x = Inf, y = 45.6, label = "Avg Precip: 45.6
inches", : Ignoring unknown parameters: `face`</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values (`geom_line()`).</code></pre>
</div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
<div class="cell-output-display">
<div>
<figure class="figure">
<p><img src="MADAproject_pt2_ruiz_files/figure-html/unnamed-chunk-17-1.png" class="img-fluid figure-img" width="672"></p>
</figure>
</div>
</div>
<div class="sourceCode cell-code" id="cb74"><pre class="sourceCode r code-with-copy"><code class="sourceCode r"><span id="cb74-1"><a href="#cb74-1" aria-hidden="true" tabindex="-1"></a><span class="co"># Save the plot</span></span>
<span id="cb74-2"><a href="#cb74-2" aria-hidden="true" tabindex="-1"></a><span class="fu">ggsave</span>(<span class="at">filename =</span> <span class="fu">here</span>(<span class="st">"results"</span>, <span class="st">"figures"</span>, <span class="st">"combined_trends_mel_precip.png"</span>), <span class="at">plot =</span> <span class="fu">last_plot</span>(), <span class="at">width =</span> <span class="dv">10</span>, <span class="at">height =</span> <span class="dv">6</span>, <span class="at">dpi =</span> <span class="dv">300</span>)</span></code><button title="Copy to Clipboard" class="code-copy-button"><i class="bi"></i></button></pre></div>
<div class="cell-output cell-output-stderr">
<pre><code>Warning: Removed 1 row containing missing values (`geom_line()`).
Removed 1 rows containing missing values (`geom_point()`).</code></pre>
</div>
</div>
<p>```</p>
</section>
</section>
</section>

</main>
<!-- /main column -->
<script id="quarto-html-after-body" type="application/javascript">
window.document.addEventListener("DOMContentLoaded", function (event) {
  const toggleBodyColorMode = (bsSheetEl) => {
    const mode = bsSheetEl.getAttribute("data-mode");
    const bodyEl = window.document.querySelector("body");
    if (mode === "dark") {
      bodyEl.classList.add("quarto-dark");
      bodyEl.classList.remove("quarto-light");
    } else {
      bodyEl.classList.add("quarto-light");
      bodyEl.classList.remove("quarto-dark");
    }
  }
  const toggleBodyColorPrimary = () => {
    const bsSheetEl = window.document.querySelector("link#quarto-bootstrap");
    if (bsSheetEl) {
      toggleBodyColorMode(bsSheetEl);
    }
  }
  toggleBodyColorPrimary();  
  const icon = "";
  const anchorJS = new window.AnchorJS();
  anchorJS.options = {
    placement: 'right',
    icon: icon
  };
  anchorJS.add('.anchored');
  const isCodeAnnotation = (el) => {
    for (const clz of el.classList) {
      if (clz.startsWith('code-annotation-')) {                     
        return true;
      }
    }
    return false;
  }
  const clipboard = new window.ClipboardJS('.code-copy-button', {
    text: function(trigger) {
      const codeEl = trigger.previousElementSibling.cloneNode(true);
      for (const childEl of codeEl.children) {
        if (isCodeAnnotation(childEl)) {
          childEl.remove();
        }
      }
      return codeEl.innerText;
    }
  });
  clipboard.on('success', function(e) {
    // button target
    const button = e.trigger;
    // don't keep focus
    button.blur();
    // flash "checked"
    button.classList.add('code-copy-button-checked');
    var currentTitle = button.getAttribute("title");
    button.setAttribute("title", "Copied!");
    let tooltip;
    if (window.bootstrap) {
      button.setAttribute("data-bs-toggle", "tooltip");
      button.setAttribute("data-bs-placement", "left");
      button.setAttribute("data-bs-title", "Copied!");
      tooltip = new bootstrap.Tooltip(button, 
        { trigger: "manual", 
          customClass: "code-copy-button-tooltip",
          offset: [0, -8]});
      tooltip.show();    
    }
    setTimeout(function() {
      if (tooltip) {
        tooltip.hide();
        button.removeAttribute("data-bs-title");
        button.removeAttribute("data-bs-toggle");
        button.removeAttribute("data-bs-placement");
      }
      button.setAttribute("title", currentTitle);
      button.classList.remove('code-copy-button-checked');
    }, 1000);
    // clear code selection
    e.clearSelection();
  });
  function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
    const config = {
      allowHTML: true,
      maxWidth: 500,
      delay: 100,
      arrow: false,
      appendTo: function(el) {
          return el.parentElement;
      },
      interactive: true,
      interactiveBorder: 10,
      theme: 'quarto',
      placement: 'bottom-start',
    };
    if (contentFn) {
      config.content = contentFn;
    }
    if (onTriggerFn) {
      config.onTrigger = onTriggerFn;
    }
    if (onUntriggerFn) {
      config.onUntrigger = onUntriggerFn;
    }
    window.tippy(el, config); 
  }
  const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
  for (var i=0; i<noterefs.length; i++) {
    const ref = noterefs[i];
    tippyHover(ref, function() {
      // use id or data attribute instead here
      let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
      try { href = new URL(href).hash; } catch {}
      const id = href.replace(/^#\/?/, "");
      const note = window.document.getElementById(id);
      return note.innerHTML;
    });
  }
  const xrefs = window.document.querySelectorAll('a.quarto-xref');
  const processXRef = (id, note) => {
    // Strip column container classes
    const stripColumnClz = (el) => {
      el.classList.remove("page-full", "page-columns");
      if (el.children) {
        for (const child of el.children) {
          stripColumnClz(child);
        }
      }
    }
    stripColumnClz(note)
    if (id === null || id.startsWith('sec-')) {
      // Special case sections, only their first couple elements
      const container = document.createElement("div");
      if (note.children && note.children.length > 2) {
        container.appendChild(note.children[0].cloneNode(true));
        for (let i = 1; i < note.children.length; i++) {
          const child = note.children[i];
          if (child.tagName === "P" && child.innerText === "") {
            continue;
          } else {
            container.appendChild(child.cloneNode(true));
            break;
          }
        }
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(container);
        }
        return container.innerHTML
      } else {
        if (window.Quarto?.typesetMath) {
          window.Quarto.typesetMath(note);
        }
        return note.innerHTML;
      }
    } else {
      // Remove any anchor links if they are present
      const anchorLink = note.querySelector('a.anchorjs-link');
      if (anchorLink) {
        anchorLink.remove();
      }
      if (window.Quarto?.typesetMath) {
        window.Quarto.typesetMath(note);
      }
      return note.innerHTML;
    }
  }
  for (var i=0; i<xrefs.length; i++) {
    const xref = xrefs[i];
    tippyHover(xref, undefined, function(instance) {
      instance.disable();
      let url = xref.getAttribute('href');
      let hash = undefined; 
      if (url.startsWith('#')) {
        hash = url;
      } else {
        try { hash = new URL(url).hash; } catch {}
      }
      if (hash) {
        const id = hash.replace(/^#\/?/, "");
        const note = window.document.getElementById(id);
        if (note !== null) {
          try {
            const html = processXRef(id, note.cloneNode(true));
            instance.setContent(html);
          } finally {
            instance.enable();
            instance.show();
          }
        } else {
          // See if we can fetch this
          fetch(url.split('#')[0])
          .then(res => res.text())
          .then(html => {
            const parser = new DOMParser();
            const htmlDoc = parser.parseFromString(html, "text/html");
            const note = htmlDoc.getElementById(id);
            if (note !== null) {
              const html = processXRef(id, note);
              instance.setContent(html);
            } 
          }).finally(() => {
            instance.enable();
            instance.show();
          });
        }
      } else {
        // See if we can fetch a full url (with no hash to target)
        // This is a special case and we should probably do some content thinning / targeting
        fetch(url)
        .then(res => res.text())
        .then(html => {
          const parser = new DOMParser();
          const htmlDoc = parser.parseFromString(html, "text/html");
          const note = htmlDoc.querySelector('main.content');
          if (note !== null) {
            // This should only happen for chapter cross references
            // (since there is no id in the URL)
            // remove the first header
            if (note.children.length > 0 && note.children[0].tagName === "HEADER") {
              note.children[0].remove();
            }
            const html = processXRef(null, note);
            instance.setContent(html);
          } 
        }).finally(() => {
          instance.enable();
          instance.show();
        });
      }
    }, function(instance) {
    });
  }
      let selectedAnnoteEl;
      const selectorForAnnotation = ( cell, annotation) => {
        let cellAttr = 'data-code-cell="' + cell + '"';
        let lineAttr = 'data-code-annotation="' +  annotation + '"';
        const selector = 'span[' + cellAttr + '][' + lineAttr + ']';
        return selector;
      }
      const selectCodeLines = (annoteEl) => {
        const doc = window.document;
        const targetCell = annoteEl.getAttribute("data-target-cell");
        const targetAnnotation = annoteEl.getAttribute("data-target-annotation");
        const annoteSpan = window.document.querySelector(selectorForAnnotation(targetCell, targetAnnotation));
        const lines = annoteSpan.getAttribute("data-code-lines").split(",");
        const lineIds = lines.map((line) => {
          return targetCell + "-" + line;
        })
        let top = null;
        let height = null;
        let parent = null;
        if (lineIds.length > 0) {
            //compute the position of the single el (top and bottom and make a div)
            const el = window.document.getElementById(lineIds[0]);
            top = el.offsetTop;
            height = el.offsetHeight;
            parent = el.parentElement.parentElement;
          if (lineIds.length > 1) {
            const lastEl = window.document.getElementById(lineIds[lineIds.length - 1]);
            const bottom = lastEl.offsetTop + lastEl.offsetHeight;
            height = bottom - top;
          }
          if (top !== null && height !== null && parent !== null) {
            // cook up a div (if necessary) and position it 
            let div = window.document.getElementById("code-annotation-line-highlight");
            if (div === null) {
              div = window.document.createElement("div");
              div.setAttribute("id", "code-annotation-line-highlight");
              div.style.position = 'absolute';
              parent.appendChild(div);
            }
            div.style.top = top - 2 + "px";
            div.style.height = height + 4 + "px";
            div.style.left = 0;
            let gutterDiv = window.document.getElementById("code-annotation-line-highlight-gutter");
            if (gutterDiv === null) {
              gutterDiv = window.document.createElement("div");
              gutterDiv.setAttribute("id", "code-annotation-line-highlight-gutter");
              gutterDiv.style.position = 'absolute';
              const codeCell = window.document.getElementById(targetCell);
              const gutter = codeCell.querySelector('.code-annotation-gutter');
              gutter.appendChild(gutterDiv);
            }
            gutterDiv.style.top = top - 2 + "px";
            gutterDiv.style.height = height + 4 + "px";
          }
          selectedAnnoteEl = annoteEl;
        }
      };
      const unselectCodeLines = () => {
        const elementsIds = ["code-annotation-line-highlight", "code-annotation-line-highlight-gutter"];
        elementsIds.forEach((elId) => {
          const div = window.document.getElementById(elId);
          if (div) {
            div.remove();
          }
        });
        selectedAnnoteEl = undefined;
      };
        // Handle positioning of the toggle
    window.addEventListener(
      "resize",
      throttle(() => {
        elRect = undefined;
        if (selectedAnnoteEl) {
          selectCodeLines(selectedAnnoteEl);
        }
      }, 10)
    );
    function throttle(fn, ms) {
    let throttle = false;
    let timer;
      return (...args) => {
        if(!throttle) { // first call gets through
            fn.apply(this, args);
            throttle = true;
        } else { // all the others get throttled
            if(timer) clearTimeout(timer); // cancel #2
            timer = setTimeout(() => {
              fn.apply(this, args);
              timer = throttle = false;
            }, ms);
        }
      };
    }
      // Attach click handler to the DT
      const annoteDls = window.document.querySelectorAll('dt[data-target-cell]');
      for (const annoteDlNode of annoteDls) {
        annoteDlNode.addEventListener('click', (event) => {
          const clickedEl = event.target;
          if (clickedEl !== selectedAnnoteEl) {
            unselectCodeLines();
            const activeEl = window.document.querySelector('dt[data-target-cell].code-annotation-active');
            if (activeEl) {
              activeEl.classList.remove('code-annotation-active');
            }
            selectCodeLines(clickedEl);
            clickedEl.classList.add('code-annotation-active');
          } else {
            // Unselect the line
            unselectCodeLines();
            clickedEl.classList.remove('code-annotation-active');
          }
        });
      }
  const findCites = (el) => {
    const parentEl = el.parentElement;
    if (parentEl) {
      const cites = parentEl.dataset.cites;
      if (cites) {
        return {
          el,
          cites: cites.split(' ')
        };
      } else {
        return findCites(el.parentElement)
      }
    } else {
      return undefined;
    }
  };
  var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
  for (var i=0; i<bibliorefs.length; i++) {
    const ref = bibliorefs[i];
    const citeInfo = findCites(ref);
    if (citeInfo) {
      tippyHover(citeInfo.el, function() {
        var popup = window.document.createElement('div');
        citeInfo.cites.forEach(function(cite) {
          var citeDiv = window.document.createElement('div');
          citeDiv.classList.add('hanging-indent');
          citeDiv.classList.add('csl-entry');
          var biblioDiv = window.document.getElementById('ref-' + cite);
          if (biblioDiv) {
            citeDiv.innerHTML = biblioDiv.innerHTML;
          }
          popup.appendChild(citeDiv);
        });
        return popup.innerHTML;
      });
    }
  }
});
</script>
</div> <!-- /content -->




</body></html>